--- refpolicy-20110726/policy/modules/services/mysql.te	2011-03-28 17:05:14.000000000 +0200
+++ refpolicy/policy/modules/services/mysql.te	2011-07-27 17:54:06.584005801 +0200
@@ -131,6 +131,10 @@
 	type_transition mysqld_t mysqld_db_t:sock_file mysqld_var_run_t;
 ')
 
+tunable_policy(`gentoo_try_dontaudit',`
+	dontaudit mysqld_t self:capability dac_read_search;
+')
+
 tunable_policy(`mysql_connect_any',`
 	corenet_tcp_connect_all_ports(mysqld_t)
 	corenet_sendrecv_all_client_packets(mysqld_t)
