--- refpolicy-20110726/policy/modules/services/ssh.te	2011-03-28 17:05:15.000000000 +0200
+++ refpolicy/policy/modules/services/ssh.te	2011-07-27 07:44:28.698004778 +0200
@@ -249,6 +249,10 @@
 corenet_tcp_bind_xserver_port(sshd_t)
 corenet_sendrecv_xserver_server_packets(sshd_t)
 
+tunable_policy(`gentoo_try_dontaudit',`
+	kernel_dontaudit_request_load_module(sshd_t)
+')
+
 tunable_policy(`ssh_sysadm_login',`
 	# Relabel and access ptys created by sshd
 	# ioctl is necessary for logout() processing for utmp entry and for w to
