--- refpolicy/policy/modules/admin/sudo.if	2011-10-25 19:19:48.000000000 +0200
+++ refpolicy/policy/modules/admin/sudo.if	2012-02-25 10:05:29.770306577 +0100
@@ -136,6 +136,7 @@
 	userdom_use_user_terminals($1_sudo_t)
 	# for some PAM modules and for cwd
 	userdom_dontaudit_search_user_home_content($1_sudo_t)
+	userdom_dontaudit_search_user_home_dirs($1_sudo_t)
 
 	ifdef(`hide_broken_symptoms', `
 		dontaudit $1_sudo_t $3:socket_class_set { read write };
