--- refpolicy/policy/modules/contrib/raid.te	2011-10-25 19:20:04.000000000 +0200
+++ refpolicy/policy/modules/contrib/raid.te	2012-02-25 09:59:58.443305869 +0100
@@ -57,6 +57,7 @@
 
 files_read_etc_files(mdadm_t)
 files_read_etc_runtime_files(mdadm_t)
+files_dontaudit_getattr_all_files(mdadm_t)
 
 fs_search_auto_mountpoints(mdadm_t)
 fs_dontaudit_list_tmpfs(mdadm_t)
@@ -70,6 +71,7 @@
 storage_read_scsi_generic(mdadm_t)
 
 term_dontaudit_list_ptys(mdadm_t)
+term_dontaudit_use_unallocated_ttys(mdadm_t)
 
 init_dontaudit_getattr_initctl(mdadm_t)
 
