--- refpolicy/policy/modules/system/mount.te	2012-02-15 20:32:44.000000000 +0100
+++ refpolicy/policy/modules/system/mount.te	2012-02-25 10:02:45.598306226 +0100
@@ -89,7 +89,8 @@
 # For reading cert files
 files_read_usr_files(mount_t)
 files_list_mnt(mount_t)
-files_dontaudit_write_root_dirs(mount_t)
+files_dontaudit_write_all_mountpoints(mount_t)
+files_dontaudit_setattr_all_mountpoints(mount_t)
 
 fs_getattr_xattr_fs(mount_t)
 fs_getattr_cifs(mount_t)
@@ -113,6 +114,7 @@
 storage_raw_write_removable_device(mount_t)
 
 term_use_all_terms(mount_t)
+term_dontaudit_manage_pty_dirs(mount_t)
 
 auth_use_nsswitch(mount_t)
 
