--- refpolicy/policy/modules/services/xserver.te	2012-02-15 20:32:44.000000000 +0100
+++ refpolicy/policy/modules/services/xserver.te	2012-02-25 10:10:44.155307249 +0100
@@ -635,6 +635,8 @@
 manage_files_pattern(xserver_t, xserver_log_t, xserver_log_t)
 logging_log_filetrans(xserver_t, xserver_log_t, file)
 
+domain_dontaudit_search_all_domains_state(xserver_t)
+
 kernel_read_system_state(xserver_t)
 kernel_read_device_sysctls(xserver_t)
 kernel_read_modprobe_sysctls(xserver_t)
