--- refpolicy/policy/modules/admin/sudo.if	2012-02-25 12:22:11.826324084 +0100
+++ refpolicy/policy/modules/admin/sudo.if	2012-02-25 12:21:58.139324055 +0100
@@ -129,6 +129,7 @@
 	seutil_libselinux_linked($1_sudo_t)
 
 	userdom_spec_domtrans_all_users($1_sudo_t)
+	userdom_create_all_users_keys($1_sudo_t)
 	userdom_manage_user_home_content_files($1_sudo_t)
 	userdom_manage_user_home_content_symlinks($1_sudo_t)
 	userdom_manage_user_tmp_files($1_sudo_t)
