--- refpolicy/policy/modules/system/lvm.te	2011-10-25 19:19:35.000000000 +0200
+++ refpolicy/policy/modules/system/lvm.te	2012-02-25 12:28:50.815324936 +0100
@@ -167,7 +167,7 @@
 # net_admin for multipath
 allow lvm_t self:capability { dac_override fowner ipc_lock sys_admin sys_nice mknod chown sys_resource sys_rawio net_admin };
 dontaudit lvm_t self:capability sys_tty_config;
-allow lvm_t self:process { sigchld sigkill sigstop signull signal };
+allow lvm_t self:process { sigchld sigkill sigstop signull signal setfscreate };
 # LVM will complain a lot if it cannot set its priority.
 allow lvm_t self:process setsched;
 allow lvm_t self:file rw_file_perms;
