--- refpolicy/policy/modules/contrib/dhcp.te	2012-03-06 21:57:11.968025583 +0100
+++ refpolicy/policy/modules/contrib/dhcp.te	2012-03-06 21:56:31.896025499 +0100
@@ -72,6 +72,7 @@
 corenet_udp_sendrecv_all_ports(dhcpd_t)
 corenet_tcp_bind_generic_node(dhcpd_t)
 corenet_udp_bind_generic_node(dhcpd_t)
+corenet_udp_bind_all_unreserved_ports(dhcpd_t) # scanning available interfaces
 corenet_tcp_bind_dhcpd_port(dhcpd_t)
 corenet_udp_bind_dhcpd_port(dhcpd_t)
 corenet_udp_bind_pxe_port(dhcpd_t)
