--- refpolicy/policy/modules/system/udev.te	2012-06-25 21:03:17.112471101 +0200
+++ refpolicypolicy/modules/system/udev.te	2012-05-28 10:26:53.624493049 +0200
@@ -74,6 +74,7 @@
 manage_dirs_pattern(udev_t, udev_var_run_t, udev_var_run_t)
 manage_files_pattern(udev_t, udev_var_run_t, udev_var_run_t)
 manage_lnk_files_pattern(udev_t, udev_var_run_t, udev_var_run_t)
+manage_sock_files_pattern(udev_t, udev_var_run_t, udev_var_run_t)
 files_pid_filetrans(udev_t, udev_var_run_t, { dir file })
 
 kernel_read_system_state(udev_t)
--- refpolicy/policy/modules/system/udev.if	2011-10-25 19:19:36.000000000 +0200
+++ refpolicypolicy/modules/system/udev.if	2012-06-25 20:50:07.404452858 +0200
@@ -132,6 +132,66 @@
 
 ########################################
 ## <summary>
+##	Do not audit attempts to get attributes
+##	of a udev netlink_kobject_uevent_socket
+## </summary>
+## <param name="domain">
+##	<summary>
+##	Domain to not audit.
+##	</summary>
+## </param>
+#
+interface(`udev_dontaudit_getattr_netlink_kobject_uevent_sockets',`
+	gen_require(`
+		type udev_t;
+	')
+
+	dontaudit $1 udev_t:netlink_kobject_uevent_socket getattr;
+')
+
+########################################
+## <summary>
+##	Do not audit attempts to get attributes
+##	of a udev unix_stream_socket.
+## </summary>
+## <param name="domain">
+##	<summary>
+##	Domain to not audit.
+##	</summary>
+## </param>
+#
+interface(`udev_dontaudit_getattr_unix_stream_sockets',`
+	gen_require(`
+		type udev_t;
+	')
+
+	dontaudit $1 udev_t:unix_stream_socket getattr;
+')
+
+
+########################################
+## <summary>
+##	Read udev rules files
+## </summary>
+## <param name="domain">
+##	<summary>
+##	Domain allowed access.
+##	</summary>
+## </param>
+#
+interface(`udev_read_rules_files',`
+	gen_require(`
+		type udev_rules_t;
+	')
+
+	files_search_etc($1) # /etc/udev/rules.d
+	udev_search_pids($1) # /run/udev/rules.d
+	read_files_pattern($1, udev_rules_t, udev_rules_t)
+')
+
+
+########################################
+## <summary>
 ##	Manage udev rules files
 ## </summary>
 ## <param name="domain">
@@ -191,6 +251,8 @@
 	allow $1 udev_tbl_t:dir list_dir_perms;
 	read_files_pattern($1, udev_tbl_t, udev_tbl_t)
 	read_lnk_files_pattern($1, udev_tbl_t, udev_tbl_t)
+	# Device table files are beneith /run/udev
+	udev_search_pids($1)
 ')
 
 ########################################
@@ -214,6 +276,49 @@
 
 ########################################
 ## <summary>
+##	Write dirs in /var/run with the udev_var_run file type
+## </summary>
+## <param name="domain">
+##	<summary>
+##	Domain allowed access.
+##	</summary>
+## </param>
+## <param name="name" optional="true">
+##	<summary>
+##	Name of the directory that the file transition will work on
+##	</summary>
+## </param>
+#
+interface(`udev_pid_filetrans_run_dirs',`
+	gen_require(`
+		type udev_var_run_t;
+	')
+
+	files_pid_filetrans($1, udev_var_run_t, dir, $2)
+')
+
+########################################
+## <summary>
+## 	Search through udev pid files and directories
+## </summary>
+## <param name="domain">
+##	<summary>
+##	Domain allowed access.
+##	</summary>
+## </param>
+#
+interface(`udev_search_pids',`
+	gen_require(`
+		type udev_var_run_t;
+	')
+
+	files_search_var_lib($1)
+	search_dirs_pattern($1, udev_var_run_t, udev_var_run_t)
+')
+
+
+########################################
+## <summary>
 ##	Create, read, write, and delete
 ##	udev pid files.
 ## </summary>
@@ -231,3 +336,23 @@
 	files_search_var_lib($1)
 	manage_files_pattern($1, udev_var_run_t, udev_var_run_t)
 ')
+
+########################################
+## <summary>
+##	Create, read, write, and delete
+##	udev run directories
+## </summary>
+## <param name="domain">
+##	<summary>
+##	Domain allowed access.
+##	</summary>
+## </param>
+#
+interface(`udev_manage_pid_dirs',`
+	gen_require(`
+		type udev_var_run_t;
+	')
+
+	files_search_var_lib($1)
+	manage_dirs_pattern($1, udev_var_run_t, udev_var_run_t)
+')
--- refpolicy/policy/modules/system/udev.fc	2012-06-25 21:03:17.237471103 +0200
+++ refpolicypolicy/modules/system/udev.fc	2012-06-24 15:29:51.794013321 +0200
@@ -31,7 +31,9 @@
 /usr/bin/udevinfo --	gen_context(system_u:object_r:udev_exec_t,s0)
 
 /var/run/PackageKit/udev(/.*)? gen_context(system_u:object_r:udev_var_run_t,s0)
-/var/run/udev(/.*)?	gen_context(system_u:object_r:udev_tbl_t,s0)
+/var/run/udev	-d	gen_context(system_u:object_r:udev_var_run_t,s0)
+/var/run/udev/rules\.d(/.*)?	gen_context(system_u:object_r:udev_tbl_t,s0)
+/var/run/udev/data(/.*)?	gen_context(system_u:object_r:udev_tbl_t,s0)
 
 ifdef(`distro_debian',`
 /var/run/xen-hotplug -d	gen_context(system_u:object_r:udev_var_run_t,s0)
