--- refpolicy/policy/modules/contrib/git.te	2012-02-15 20:32:44.000000000 +0100
+++ refpolicypolicy/modules/contrib/git.te	2012-07-10 18:50:41.770309289 +0200
@@ -108,6 +108,8 @@
 corenet_tcp_sendrecv_git_port(git_session_t)
 corenet_sendrecv_git_server_packets(git_session_t)
 
+auth_use_nsswitch(git_session_t)
+
 userdom_use_user_terminals(git_session_t)
 
 tunable_policy(`git_session_send_syslog_msg',`
@@ -135,6 +137,8 @@
 read_files_pattern(git_system_t, git_sys_content_t, git_sys_content_t)
 files_search_var_lib(git_system_t)
 
+auth_use_nsswitch(git_system_t)
+
 logging_send_syslog_msg(git_system_t)
 
 tunable_policy(`git_system_enable_homedirs',`
@@ -221,6 +225,4 @@
 
 fs_search_auto_mountpoints(git_daemon)
 
-auth_use_nsswitch(git_daemon)
-
 miscfiles_read_localization(git_daemon)
