--- refpolicy/policy/modules/contrib/mozilla.te	2012-02-25 10:15:51.219307903 +0100
+++ refpolicy/policy/modules/contrib/mozilla.te	2012-02-25 10:12:31.693307477 +0100
@@ -148,6 +148,7 @@
 files_read_var_symlinks(mozilla_t)
 files_dontaudit_getattr_boot_dirs(mozilla_t)
 
+fs_dontaudit_getattr_all_fs(mozilla_t)
 fs_search_auto_mountpoints(mozilla_t)
 fs_list_inotifyfs(mozilla_t)
 fs_rw_tmpfs_files(mozilla_t)
