--- refpolicy/policy/modules/system/selinuxutil.te	2012-02-15 20:32:44.000000000 +0100
+++ refpolicy/policy/modules/system/selinuxutil.te	2012-02-25 10:05:12.238306540 +0100
@@ -232,6 +232,8 @@
 allow newrole_t self:unix_stream_socket { create_stream_socket_perms connectto };
 allow newrole_t self:netlink_audit_socket { create_netlink_socket_perms nlmsg_relay };
 
+dontaudit newrole_t self:capability dac_read_search;
+
 read_files_pattern(newrole_t, default_context_t, default_context_t)
 read_lnk_files_pattern(newrole_t, default_context_t, default_context_t)
 
