--- refpolicy/policy/modules/contrib/asterisk.te	2012-02-25 11:21:36.364316323 +0100
+++ refpolicy/policy/modules/contrib/asterisk.te	2012-02-25 11:21:24.326316299 +0100
@@ -46,7 +46,7 @@
 allow asterisk_t self:fifo_file rw_fifo_file_perms;
 allow asterisk_t self:sem create_sem_perms;
 allow asterisk_t self:shm create_shm_perms;
-allow asterisk_t self:unix_stream_socket connectto;
+allow asterisk_t self:unix_stream_socket { connectto listen };
 allow asterisk_t self:tcp_socket create_stream_socket_perms;
 allow asterisk_t self:udp_socket create_socket_perms;
 
