--- refpolicy/policy/modules/contrib/clamav.if	2011-10-25 19:19:58.000000000 +0200
+++ refpolicy/policy/modules/contrib/clamav.if	2012-03-01 20:39:36.425379112 +0100
@@ -49,12 +49,12 @@
 #
 interface(`clamav_append_log',`
 	gen_require(`
-		type clamav_log_t;
+		type clamav_var_log_t;
 	')
 
 	logging_search_logs($1)
-	allow $1 clamav_log_t:dir list_dir_perms;
-	append_files_pattern($1, clamav_log_t, clamav_log_t)
+	allow $1 clamav_var_log_t:dir list_dir_perms;
+	append_files_pattern($1, clamav_var_log_t, clamav_var_log_t)
 ')
 
 ########################################
