--- refpolicy/policy/modules/contrib/zarafa.te	2011-10-25 19:20:09.000000000 +0200
+++ refpolicypolicy/modules/contrib/zarafa.te	2012-07-10 18:58:26.219302399 +0200
@@ -41,6 +41,8 @@
 manage_files_pattern(zarafa_deliver_t, zarafa_deliver_tmp_t, zarafa_deliver_tmp_t)
 files_tmp_filetrans(zarafa_deliver_t, zarafa_deliver_tmp_t, { file dir })
 
+auth_use_nsswitch(zarafa_deliver_t)
+
 ########################################
 #
 # zarafa_gateway local policy
@@ -57,6 +59,8 @@
 corenet_tcp_bind_generic_node(zarafa_gateway_t)
 corenet_tcp_bind_pop_port(zarafa_gateway_t)
 
+auth_use_nsswitch(zarafa_gateway_t)
+
 #######################################
 #
 # zarafa-ical local policy
@@ -72,6 +76,8 @@
 corenet_tcp_bind_generic_node(zarafa_ical_t)
 corenet_tcp_bind_http_cache_port(zarafa_ical_t)
 
+auth_use_nsswitch(zarafa_ical_t)
+
 ######################################
 #
 # zarafa-monitor local policy
@@ -79,6 +85,8 @@
 
 allow zarafa_monitor_t self:capability chown;
 
+auth_use_nsswitch(zarafa_monitor_t)
+
 ########################################
 #
 # zarafa_server local policy
@@ -107,6 +115,8 @@
 
 files_read_usr_files(zarafa_server_t)
 
+auth_use_nsswitch(zarafa_server_t)
+
 logging_send_syslog_msg(zarafa_server_t)
 logging_send_audit_msgs(zarafa_server_t)
 
@@ -136,6 +146,8 @@
 corenet_tcp_sendrecv_all_ports(zarafa_spooler_t)
 corenet_tcp_connect_smtp_port(zarafa_spooler_t)
 
+auth_use_nsswitch(zarafa_spooler_t)
+
 ########################################
 #
 # zarafa domains local policy
@@ -156,6 +168,4 @@
 
 files_read_etc_files(zarafa_domain)
 
-auth_use_nsswitch(zarafa_domain)
-
 miscfiles_read_localization(zarafa_domain)
