--- refpolicy/policy/modules/system/udev.if	2011-10-25 19:19:36.000000000 +0200
+++ refpolicy/policy/modules/system/udev.if	2012-02-25 12:24:33.000324386 +0100
@@ -132,6 +132,25 @@
 
 ########################################
 ## <summary>
+##	Read udev rules files
+## </summary>
+## <param name="domain">
+##	<summary>
+##	Domain allowed access.
+##	</summary>
+## </param>
+#
+interface(`udev_read_rules_files',`
+	gen_require(`
+		type udev_rules_t;
+	')
+
+	read_files_pattern($1, udev_rules_t, udev_rules_t)
+')
+
+
+########################################
+## <summary>
 ##	Manage udev rules files
 ## </summary>
 ## <param name="domain">
