--- refpolicy/policy/modules/system/selinuxutil.te	2012-02-26 15:39:41.084533791 +0100
+++ refpolicy/policy/modules/system/selinuxutil.te	2012-02-26 15:38:59.897533703 +0100
@@ -13,9 +13,11 @@
 attribute can_relabelto_binary_policy;
 
 attribute_role newrole_roles;
+role newrole_roles types newrole_t;
 
 attribute_role run_init_roles;
 role system_r types run_init_t;
+role run_init_roles types run_init_t;
 
 attribute_role semanage_roles;
 roleattribute system_r semanage_roles;
