--- refpolicy/policy/modules/contrib/mysql.if	2012-07-31 19:42:22.001874623 +0200
+++ refpolicy/policy/modules/contrib/mysql.if	2012-07-31 19:42:11.520874601 +0200
@@ -313,6 +313,24 @@
 
 #######################################
 ## <summary>
+##	Set the attributes of the MySQL run directories
+## </summary>
+## <param name="domain">
+##	<summary>
+##	Domain allowed access
+##	</summary>
+## </param>
+#
+interface(`mysql_setattr_run_dirs',`
+	gen_require(`
+		type mysqld_var_run_t;
+	')
+
+	setattr_dirs_pattern($1, mysqld_var_run_t, mysqld_var_run_t)
+')
+
+#######################################
+## <summary>
 ##	Create MySQL run directories
 ## </summary>
 ## <param name="domain">
