--- refpolicy/policy/modules/contrib/dbus.te	2012-07-25 20:33:04.000000000 +0200
+++ refpolicy/policy/modules/contrib/dbus.te	2012-08-14 18:34:54.253277137 +0200
@@ -35,6 +35,7 @@
 
 type system_dbusd_var_run_t;
 files_pid_file(system_dbusd_var_run_t)
+init_daemon_run_dir(system_dbusd_var_run_t, "dbus")
 
 ifdef(`enable_mcs',`
 	init_ranged_system_domain(system_dbusd_t, dbusd_exec_t, s0 - mcs_systemhigh)
--- refpolicy/policy/modules/contrib/dbus.if	2012-08-14 18:38:46.982286463 +0200
+++ refpolicy/policy/modules/contrib/dbus.if	2012-08-14 18:34:54.253277137 +0200
@@ -393,6 +393,50 @@
 
 ########################################
 ## <summary>
+##	Create resources in /run or /var/run with the system_dbusd_var_run_t
+##	label. This method is deprecated in favor of the init_daemon_run_dir
+##	call.
+## </summary>
+## <param name="domain">
+##	<summary>
+##	Domain allowed access
+##	</summary>
+## </param>
+## <param name="class">
+##	<summary>
+##	Classes supported for the created resources
+##	</summary>
+## </param>
+## <param name="filename" optional="true">
+##	<summary>
+##	Optional file name used for the resource
+##	</summary>
+## </param>
+#
+interface(`dbus_generic_pid_filetrans_system_dbusd_var_run',`
+	refpolicywarn(`$0($*) has been deprecated.')
+')
+
+########################################
+## <summary>
+##	Create directories with the system_dbusd_var_run_t label
+## </summary>
+## <param name="domain">
+##	<summary>
+##	Domain allowed access
+##	</summary>
+## </param>
+#
+interface(`dbus_create_system_dbusd_var_run_dirs',`
+	gen_require(`
+		type system_dbusd_var_run_t;
+	')
+
+	create_dirs_pattern($1, system_dbusd_var_run_t, system_dbusd_var_run_t)
+')
+
+########################################
+## <summary>
 ##	Allow unconfined access to the system DBUS.
 ## </summary>
 ## <param name="domain">
