--- refpolicy/policy/modules/kernel/devices.fc	2012-08-24 18:39:03.792740611 +0200
+++ refpolicy/policy/modules/kernel/devices.fc	2012-08-24 18:37:59.878740475 +0200
@@ -126,8 +126,8 @@
 /dev/vrtpanel		-c	gen_context(system_u:object_r:mouse_device_t,s0)
 /dev/vttuner		-c	gen_context(system_u:object_r:v4l_device_t,s0)
 /dev/vtx.*		-c	gen_context(system_u:object_r:v4l_device_t,s0)
-/dev/watchdog		-c	gen_context(system_u:object_r:watchdog_device_t,s0)
-/dev/winradio.		-c	gen_context(system_u:object_r:v4l_device_t,s0)
+/dev/watchdog.*		-c	gen_context(system_u:object_r:watchdog_device_t,s0)
+/dev/winradio.*		-c	gen_context(system_u:object_r:v4l_device_t,s0)
 /dev/z90crypt		-c	gen_context(system_u:object_r:crypt_device_t,s0)
 /dev/zero		-c	gen_context(system_u:object_r:zero_device_t,s0)
 
