diff -uNr refpolicy/policy/modules/system/udev.te refpolicy/policy/modules/system/udev.te
--- refpolicy/policy/modules/system/udev.te	2012-07-26 21:09:59.529963767 +0200
+++ refpolicy/policy/modules/system/udev.te	2012-07-26 21:09:59.600963767 +0200
@@ -175,6 +175,7 @@
 	# during boot, init scripts use /dev/.rcsysinit
 	# existance to determine if we are in early booting
 	init_getattr_script_status_files(udev_t)
+	init_domtrans_script(udev_t)
 ')
 
 ifdef(`distro_redhat',`
