diff -uNr refpolicy/policy/modules/roles/unprivuser.te refpolicy/policy/modules/roles/unprivuser.te
--- refpolicy/policy/modules/roles/unprivuser.te	2012-07-26 21:10:00.821963769 +0200
+++ refpolicy/policy/modules/roles/unprivuser.te	2012-07-26 21:10:00.956963769 +0200
@@ -110,6 +110,10 @@
 	')
 
 	optional_policy(`
+		pan_role(user_r, user_t)
+	')
+
+	optional_policy(`
 		postgresql_role(user_r, user_t)
 	')
 
