--- refpolicy/modules/services/nagios.te	2011-08-16 23:29:40.342473953 +0000
+++ refpolicy/modules/services/nagios.te	2011-08-16 23:27:13.000000000 +0000
@@ -310,6 +310,7 @@
 # needed by ioctl()
 allow nagios_checkdisk_plugin_t self:capability { sys_admin sys_rawio };
 
+files_getattr_all_mountpoints(nagios_checkdisk_plugin_t)
 files_read_etc_runtime_files(nagios_checkdisk_plugin_t)
 
 fs_getattr_all_fs(nagios_checkdisk_plugin_t)
