--- refpolicy.orig/policy/modules/services/rpc.te	2011-08-15 18:40:34.060000919 +0200
+++ refpolicy/policy/modules/services/rpc.te	2011-08-15 18:39:33.866000917 +0200
@@ -113,7 +113,7 @@
 #
 
 allow nfsd_t self:capability { dac_override dac_read_search sys_admin sys_resource };
-
+allow nfsd_t self:udp_socket listen;
 allow nfsd_t exports_t:file read_file_perms;
 allow nfsd_t { nfsd_rw_t nfsd_ro_t }:dir list_dir_perms;
 
