diff -uNr refpolicy/policy/modules/roles/unprivuser.te refpolicy/policy/modules/roles/unprivuser.te
--- refpolicy/policy/modules/roles/unprivuser.te	2012-07-26 21:10:01.418963771 +0200
+++ refpolicy/policy/modules/roles/unprivuser.te	2012-07-26 21:10:01.622963771 +0200
@@ -70,6 +70,10 @@
 	')
 
 	optional_policy(`
+		gorg_role(user_r, user_t)
+	')
+
+	optional_policy(`
 		gpg_role(user_r, user_t)
 	')
 
