--- refpolicy/policy/modules/system/sysnetwork.te	2012-02-15 20:32:44.000000000 +0100
+++ refpolicy/policy/modules/system/sysnetwork.te	2012-02-25 10:18:14.535308210 +0100
@@ -338,6 +338,10 @@
 ')
 
 optional_policy(`
+	networkmanager_dontaudit_use_wpa_cli_fds(ifconfig_t)
+')
+
+optional_policy(`
 	nis_use_ypbind(ifconfig_t)
 ')
 
--- refpolicy/policy/modules/system/sysnetwork.if	2011-11-15 15:28:16.000000000 +0100
+++ refpolicy/policy/modules/system/sysnetwork.if	2012-02-25 10:14:07.151307681 +0100
@@ -66,6 +66,27 @@
 
 ########################################
 ## <summary>
+##	Do not audit attempts to read/write to
+##	the dhcp unix stream socket descriptors.
+## </summary>
+## <param name="domain">
+##	<summary>
+##	Domain to not audit.
+##	</summary>
+## </param>
+#
+interface(`sysnet_dontaudit_rw_dhcpc_unix_stream_sockets',`
+	gen_require(`
+		type dhcpc_t;
+	')
+
+	dontaudit $1 dhcpc_t:unix_stream_socket { read write };
+')
+
+
+
+########################################
+## <summary>
 ##	Send a SIGCHLD signal to the dhcp client.
 ## </summary>
 ## <param name="domain">
