--- refpolicy/policy/modules/contrib/mta.if	2011-10-25 19:20:02.000000000 +0200
+++ refpolicy/policy/modules/contrib/mta.if	2012-02-25 10:11:01.631307286 +0100
@@ -362,6 +362,8 @@
 	allow mta_user_agent $1:fd use;
 	allow mta_user_agent $1:process sigchld;
 	allow mta_user_agent $1:fifo_file rw_fifo_file_perms;
+
+	dontaudit mta_user_agent $1:unix_stream_socket rw_socket_perms;
 ')
 
 ########################################
