--- refpolicy/policy/modules/kernel/files.if	2011-11-15 15:28:16.000000000 +0100
+++ refpolicy/policy/modules/kernel/files.if	2012-02-25 10:02:13.796306159 +0100
@@ -1482,6 +1482,42 @@
 
 ########################################
 ## <summary>
+##	Do not audit write attempts on mount points.
+## </summary>
+## <param name="domain">
+##	<summary>
+##	Domain to ignore write attempts from
+##	</summary>
+## </param>
+#
+interface(`files_dontaudit_write_all_mountpoints',`
+	gen_require(`
+		attribute mountpoint;
+	')
+
+	dontaudit $1 mountpoint:dir write;
+')
+
+########################################
+## <summary>
+##	Do not audit setattr attempts on mount points.
+## </summary>
+## <param name="domain">
+##	<summary>
+##	Domain to ignore setattr attempts from
+##	</summary>
+## </param>
+#
+interface(`files_dontaudit_setattr_all_mountpoints',`
+	gen_require(`
+		attribute mountpoint;
+	')
+
+	dontaudit $1 mountpoint:dir setattr;
+')
+
+########################################
+## <summary>
 ##	List the contents of the root directory.
 ## </summary>
 ## <param name="domain">
