--- refpolicy.orig/policy/modules/admin/portage.if	2011-08-14 21:42:13.088001300 +0200
+++ refpolicy/policy/modules/admin/portage.if	2011-08-14 21:42:00.192001300 +0200
@@ -333,3 +333,22 @@
 
 	dontaudit $1 portage_tmp_t:file rw_file_perms;
 ')
+
+########################################
+## <summary>
+##     Do not audit attempts to inherit a
+##     portage file descriptor.
+## </summary>
+## <param name="domain">
+##     <summary>
+##     Domain to not audit.
+##     </summary>
+## </param>
+#
+interface(`gentoo_portage_dontaudit_use_fds',`
+       gen_require(`
+               type portage_t;
+       ')
+
+       dontaudit $1 portage_t:fd use;
+')
