diff -uNr refpolicy/policy/modules/system/logging.fc refpolicy/policy/modules/system/logging.fc
--- refpolicy/policy/modules/system/logging.fc	2012-07-26 21:09:59.528963767 +0200
+++ refpolicy/policy/modules/system/logging.fc	2012-07-26 21:10:02.024963771 +0200
@@ -69,7 +69,7 @@
 
 /var/spool/audit(/.*)?		gen_context(system_u:object_r:audit_spool_t,mls_systemhigh)
 /var/spool/bacula/log(/.*)? 	gen_context(system_u:object_r:var_log_t,s0)
-/var/spool/postfix/pid	-d	gen_context(system_u:object_r:var_run_t,s0)
+#/var/spool/postfix/pid	-d	gen_context(system_u:object_r:var_run_t,s0)
 /var/spool/plymouth/boot\.log	gen_context(system_u:object_r:var_log_t,mls_systemhigh)
 /var/spool/rsyslog(/.*)? 	gen_context(system_u:object_r:var_log_t,s0)
 
diff -uNr refpolicy/policy/modules/system/logging.te refpolicy/policy/modules/system/logging.te
--- refpolicy/policy/modules/system/logging.te	2012-07-26 21:09:59.528963767 +0200
+++ refpolicy/policy/modules/system/logging.te	2012-07-26 21:10:02.025963771 +0200
@@ -439,6 +439,8 @@
 # /initrd is not umounted before minilog starts
 files_dontaudit_search_isid_type_dirs(syslogd_t)
 files_read_kernel_symbol_table(syslogd_t)
+files_rw_var_lib_dirs(syslogd_t)
+files_var_lib_filetrans(syslogd_t, syslogd_var_lib_t, file)
 
 fs_getattr_all_fs(syslogd_t)
 fs_search_auto_mountpoints(syslogd_t)
