--- refpolicy/policy/modules/system/userdomain.te	2011-07-26 14:10:40.000000000 +0200
+++ refpolicy/policy/modules/system/userdomain.te	2011-08-11 22:50:29.005000656 +0200
@@ -94,3 +94,9 @@
 type user_tty_device_t alias { staff_tty_device_t sysadm_tty_device_t secadm_tty_device_t auditadm_tty_device_t unconfined_tty_device_t };
 dev_node(user_tty_device_t)
 ubac_constrained(user_tty_device_t)
+
+type xdg_data_home_t;
+userdom_user_home_content(xdg_data_home_t)
+
+type xdg_config_home_t;
+userdom_user_home_content(xdg_config_home_t)
--- refpolicy/policy/modules/system/userdomain.if	2011-08-22 21:32:51.225000501 +0200
+++ refpolicy/policy/modules/system/userdomain.if	2011-08-22 21:32:15.902000501 +0200
@@ -243,6 +243,29 @@
 	filetrans_pattern($2, user_home_dir_t, user_home_t, { dir file lnk_file sock_file fifo_file })
 	files_list_home($2)
 
+	# full control over the xdg_* directories
+	manage_dirs_pattern($2, xdg_config_home_t, xdg_config_home_t)
+	manage_files_pattern($2, xdg_config_home_t, xdg_config_home_t)
+	manage_lnk_files_pattern($2, xdg_config_home_t, xdg_config_home_t)
+	manage_sock_files_pattern($2, xdg_config_home_t, xdg_config_home_t)
+	manage_fifo_files_pattern($2, xdg_config_home_t, xdg_config_home_t)
+	relabel_dirs_pattern($2, xdg_config_home_t, xdg_config_home_t)
+	relabel_files_pattern($2, xdg_config_home_t, xdg_config_home_t)
+	relabel_lnk_files_pattern($2, xdg_config_home_t, xdg_config_home_t)
+	relabel_sock_files_pattern($2, xdg_config_home_t, xdg_config_home_t)
+	relabel_fifo_files_pattern($2, xdg_config_home_t, xdg_config_home_t)
+
+	manage_dirs_pattern($2, xdg_data_home_t, xdg_data_home_t)
+	manage_files_pattern($2, xdg_data_home_t, xdg_data_home_t)
+	manage_lnk_files_pattern($2, xdg_data_home_t, xdg_data_home_t)
+	manage_sock_files_pattern($2, xdg_data_home_t, xdg_data_home_t)
+	manage_fifo_files_pattern($2, xdg_data_home_t, xdg_data_home_t)
+	relabel_dirs_pattern($2, xdg_data_home_t, xdg_data_home_t)
+	relabel_files_pattern($2, xdg_data_home_t, xdg_data_home_t)
+	relabel_lnk_files_pattern($2, xdg_data_home_t, xdg_data_home_t)
+	relabel_sock_files_pattern($2, xdg_data_home_t, xdg_data_home_t)
+	relabel_fifo_files_pattern($2, xdg_data_home_t, xdg_data_home_t)
+
 	# cjp: this should probably be removed:
 	allow $2 user_home_dir_t:dir { manage_dir_perms relabel_dir_perms };
 
@@ -3197,3 +3220,84 @@
 
 	allow $1 userdomain:dbus send_msg;
 ')
+
+########################################
+## <summary>
+##	Manage the xdg data home files
+## </summary>
+## <param name="domain">
+##	<summary>
+##	Domain allowed access.
+##	</summary>
+## </param>
+#
+interface(`userdom_manage_xdg_data_home',`
+	gen_require(`
+		type xdg_data_home_t;	
+	')
+
+	manage_files_pattern($1, xdg_data_home_t, xdg_data_home_t)
+	manage_dirs_pattern($1, xdg_data_home_t, xdg_data_home_t)
+')
+
+########################################
+## <summary>
+##	Read the xdg data home files
+## </summary>
+## <param name="domain">
+##	<summary>
+##	Domain allowed access.
+##	</summary>
+## </param>
+#
+interface(`userdom_read_xdg_data_home',`
+	gen_require(`
+		type xdg_data_home_t;	
+		type user_home_dir_t;
+		type user_home_t;
+	')
+
+	search_dirs_pattern($1, user_home_dir_t, user_home_t)
+	list_dirs_pattern($1, user_home_t, xdg_data_home_t)
+	read_files_pattern($1, xdg_data_home_t, xdg_data_home_t)
+')
+
+########################################
+## <summary>
+##	Manage the xdg config home files
+## </summary>
+## <param name="domain">
+##	<summary>
+##	Domain allowed access.
+##	</summary>
+## </param>
+#
+interface(`userdom_manage_xdg_config_home',`
+	gen_require(`
+		type xdg_config_home_t;	
+	')
+
+	manage_files_pattern($1, xdg_config_home_t, xdg_config_home_t)
+	manage_dirs_pattern($1, xdg_config_home_t, xdg_config_home_t)
+')
+
+########################################
+## <summary>
+##	Read the xdg config home files
+## </summary>
+## <param name="domain">
+##	<summary>
+##	Domain allowed access.
+##	</summary>
+## </param>
+#
+interface(`userdom_read_xdg_config_home',`
+	gen_require(`
+		type xdg_config_home_t;	
+		type user_home_dir_t;
+	')
+
+	list_dirs_pattern($1, user_home_dir_t, xdg_config_home_t)
+	read_files_pattern($1, xdg_config_home_t, xdg_config_home_t)
+')
+
--- refpolicy/policy/modules/system/userdomain.fc	2011-03-28 17:05:16.000000000 +0200
+++ refpolicy/policy/modules/system/userdomain.fc	2011-08-11 22:50:28.991000656 +0200
@@ -1,4 +1,6 @@
 HOME_DIR	-d	gen_context(system_u:object_r:user_home_dir_t,s0-mls_systemhigh)
 HOME_DIR/.+		gen_context(system_u:object_r:user_home_t,s0)
+HOME_DIR/\.local/share(/.*)?	gen_context(system_u:object_r:xdg_data_home_t,s0)
+HOME_DIR/\.config(/.*)?		gen_context(system_u:object_r:xdg_config_home_t,s0)
 
 /tmp/gconfd-USER -d	gen_context(system_u:object_r:user_tmp_t,s0)
