--- refpolicy-20110726/policy/modules/services/dbus.if	2011-03-28 17:05:14.000000000 +0200
+++ refpolicy/policy/modules/services/dbus.if	2011-08-10 18:50:31.376000360 +0200
@@ -150,6 +150,7 @@
 	term_use_all_terms($1_dbusd_t)
 
 	userdom_read_user_home_content_files($1_dbusd_t)
+	userdom_read_xdg_data_home($1_dbusd_t)
 
 	ifdef(`hide_broken_symptoms', `
 		dontaudit $3 $1_dbusd_t:netlink_selinux_socket { read write };
