--- refpolicy-20110726/policy/modules/apps/mozilla.te	2011-07-26 14:10:39.000000000 +0200
+++ refpolicy/policy/modules/apps/mozilla.te	2011-08-09 23:05:18.552001714 +0200
@@ -165,6 +172,7 @@
 # Browse the web, connect to printer
 sysnet_dns_name_resolve(mozilla_t)
 
+userdom_read_xdg_data_home(mozilla_t)
 userdom_use_user_ptys(mozilla_t)
 
 xserver_user_x_domain_template(mozilla, mozilla_t, mozilla_tmpfs_t)
