diff -uNr refpolicy/policy/modules/contrib/telnet.te refpolicy/policy/modules/contrib/telnet.te
--- refpolicy/policy/modules/contrib/telnet.te	2012-07-26 21:09:59.565963767 +0200
+++ refpolicy/policy/modules/contrib/telnet.te	2012-07-26 21:10:02.693963773 +0200
@@ -81,11 +81,13 @@
 
 seutil_read_config(telnetd_t)
 
-remotelogin_domtrans(telnetd_t)
-
 userdom_search_user_home_dirs(telnetd_t)
 userdom_setattr_user_ptys(telnetd_t)
 
+optional_policy(`
+	remotelogin_domtrans(telnetd_t)
+')
+
 tunable_policy(`use_nfs_home_dirs',`
 	fs_search_nfs(telnetd_t)
 ')
