--- refpolicy/policy/modules/admin/portage.te	2011-09-03 12:45:13.786002151 +0200
+++ refpolicy/policy/modules/admin/portage.te	2011-09-03 12:44:17.655002150 +0200
@@ -122,6 +122,8 @@
 
 # seems to be ok without this
 init_dontaudit_read_script_status_files(gcc_config_t)
+# Allow gcc-config to call rc (without transitioning)
+init_rc_exec(gcc_config_t)
 
 libs_read_lib_files(gcc_config_t)
 libs_domtrans_ldconfig(gcc_config_t)
