--- refpolicy/policy/modules/admin/portage.te	2011-09-03 16:12:43.449002507 +0200
+++ refpolicy/policy/modules/admin/portage.te	2011-09-03 16:12:00.345002506 +0200
@@ -222,6 +222,11 @@
 ')
 
 optional_policy(`
+	cron_system_entry(portage_t, portage_exec_t)
+	cron_system_entry(portage_fetch_t, portage_fetch_exec_t)
+')
+
+optional_policy(`
 	modutils_domtrans_depmod(portage_t)
 	modutils_domtrans_update_mods(portage_t)
 	#dontaudit update_modules_t portage_tmp_t:dir search_dir_perms;
