--- refpolicy/policy/modules/kernel/files.if	2012-02-25 11:33:43.198317876 +0100
+++ refpolicy/policy/modules/kernel/files.if	2012-02-25 11:32:44.888317751 +0100
@@ -2836,6 +2836,26 @@
 
 ########################################
 ## <summary>
+##	Create, read, write, and delete symbolic links in
+##	/etc that are dynamically created on boot.
+## </summary>
+## <param name="domain">
+##	<summary>
+##	Domain allowed access.
+##	</summary>
+## </param>
+## <rolecap/>
+#
+interface(`files_manage_etc_runtime_lnk_files',`
+	gen_require(`
+		type etc_t, etc_runtime_t;
+	')
+
+	manage_lnk_files_pattern($1, { etc_t etc_runtime_t }, etc_runtime_t)
+')
+
+########################################
+## <summary>
 ##	Create, etc runtime objects with an automatic
 ##	type transition.
 ## </summary>
--- refpolicy/policy/modules/kernel/files.fc	2011-11-16 20:51:52.000000000 +0100
+++ refpolicy/policy/modules/kernel/files.fc	2012-02-25 11:32:44.888317751 +0100
@@ -73,7 +73,7 @@
 ifdef(`distro_gentoo', `
 /etc/profile\.env	--	gen_context(system_u:object_r:etc_runtime_t,s0)
 /etc/csh\.env		--	gen_context(system_u:object_r:etc_runtime_t,s0)
-/etc/env\.d/.*		--	gen_context(system_u:object_r:etc_runtime_t,s0)
+/etc/env\.d(/.*)?		gen_context(system_u:object_r:etc_runtime_t,s0)
 ')
 
 ifdef(`distro_redhat',`
