diff -uNr refpolicy/policy/modules/contrib/portage.te refpolicy/policy/modules/contrib/portage.te
--- refpolicy/policy/modules/contrib/portage.te	2012-07-26 21:10:03.024963775 +0200
+++ refpolicy/policy/modules/contrib/portage.te	2012-07-26 21:10:04.043963777 +0200
@@ -12,6 +12,21 @@
 ## </desc>
 gen_tunable(portage_use_nfs, false)
 
+## <desc>
+## <p>
+## (deprecated) support for dontaudit tryouts
+## </p>
+## </desc>
+gen_tunable(gentoo_try_dontaudit, false)
+
+## <desc>
+## <p>
+## (deprecated) support for fixes
+## </p>
+## </desc>
+gen_tunable(gentoo_wait_requests, false)
+
+
 attribute_role portage_roles;
 
 type gcc_config_t;
