--- refpolicy/policy/modules/contrib/portage.te	2012-02-25 12:32:43.128325431 +0100
+++ refpolicy/policy/modules/contrib/portage.te	2012-02-25 12:31:51.479325322 +0100
@@ -261,6 +261,8 @@
 allow portage_fetch_t portage_tmp_t:dir manage_dir_perms;
 allow portage_fetch_t portage_tmp_t:file manage_file_perms;
 
+allow portage_fetch_t portage_devpts_t:chr_file { rw_chr_file_perms setattr };
+
 read_files_pattern(portage_fetch_t, portage_conf_t, portage_conf_t)
 
 manage_dirs_pattern(portage_fetch_t, portage_ebuild_t, portage_ebuild_t)
