diff -uNr refpolicy/policy/modules/system/init.if refpolicy/policy/modules/system/init.if
--- refpolicy/policy/modules/system/init.if	2012-07-26 21:10:04.040963777 +0200
+++ refpolicy/policy/modules/system/init.if	2012-07-26 21:10:04.643963777 +0200
@@ -814,14 +814,6 @@
 	files_list_etc($1)
 	spec_domtrans_pattern($1, initrc_exec_t, initrc_t)
 
-	ifdef(`distro_gentoo',`
-		gen_require(`
-			type rc_exec_t;
-		')
-
-		domtrans_pattern($1, rc_exec_t, initrc_t)
-	')
-
 	ifdef(`enable_mcs',`
 		range_transition $1 initrc_exec_t:process s0;
 	')
