--- refpolicy.orig/policy/modules/services/asterisk.te	2011-08-15 17:23:30.257000787 +0200
+++ refpolicy/policy/modules/services/asterisk.te	2011-08-15 17:21:50.190000785 +0200
@@ -39,7 +39,7 @@
 #
 
 # dac_override for /var/run/asterisk
-allow asterisk_t self:capability { dac_override setgid setuid sys_nice net_admin };
+allow asterisk_t self:capability { dac_override setgid setuid sys_nice net_admin chown };
 dontaudit asterisk_t self:capability sys_tty_config;
 allow asterisk_t self:process { getsched setsched signal_perms getcap setcap };
 allow asterisk_t self:fifo_file rw_fifo_file_perms;
