--- refpolicy/policy/modules/kernel/domain.te	2011-07-26 12:10:40.000000000 +0000
+++ refpolicy/policy/modules/kernel/domain.te	2011-08-17 20:57:35.000000000 +0000
@@ -86,6 +86,7 @@
 allow domain self:dir list_dir_perms;
 allow domain self:lnk_file { read_lnk_file_perms lock ioctl };
 allow domain self:file rw_file_perms;
+dontaudit domain self:udp_socket listen;
 kernel_read_proc_symlinks(domain)
 # Every domain gets the key ring, so we should default
 # to no one allowed to look at it; afs kernel support creates
