--- refpolicy/policy/modules/services/asterisk.te	2011-08-22 20:11:41.905000361 +0200
+++ refpolicy/policy/modules/services/asterisk.te	2011-08-22 20:08:35.839000357 +0200
@@ -45,7 +45,7 @@
 allow asterisk_t self:fifo_file rw_fifo_file_perms;
 allow asterisk_t self:sem create_sem_perms;
 allow asterisk_t self:shm create_shm_perms;
-allow asterisk_t self:unix_stream_socket { connectto listen };
+allow asterisk_t self:unix_stream_socket { connectto listen accept };
 allow asterisk_t self:tcp_socket create_stream_socket_perms;
 allow asterisk_t self:udp_socket create_socket_perms;
 
--- refpolicy/policy/modules/services/asterisk.if	2011-08-22 20:11:41.910000361 +0200
+++ refpolicy/policy/modules/services/asterisk.if	2011-08-22 20:11:15.639000361 +0200
@@ -95,3 +95,46 @@
 	can_exec($1, asterisk_exec_t)
 	asterisk_stream_connect($1)
 ')
+
+#######################################
+## <summary>
+##   Allow changing the attributes of the asterisk log files and directories
+## </summary>
+## <param name="domain">
+##   <summary>
+##     Domain allowed to change the attributes of the asterisk log files and
+##     directories
+##   </summary>
+## </param>
+#
+interface(`asterisk_setattr_logs',`
+	gen_require(`
+		type asterisk_log_t;
+	')
+
+	setattr_files_pattern($1, asterisk_log_t, asterisk_log_t)
+	setattr_dirs_pattern($1, asterisk_log_t, asterisk_log_t)
+
+	logging_search_logs($1)
+')
+
+#######################################
+## <summary>
+##   Allow changing the attributes of the asterisk PID files
+## </summary>
+## <param name="domain">
+##   <summary>
+##     Domain allowed to change the attributes of the asterisk PID files
+##   </summary>
+## </param>
+#
+interface(`asterisk_setattr_pid_files',`
+	gen_require(`
+		type asterisk_var_run_t;
+	')
+
+	setattr_files_pattern($1, asterisk_var_run_t, asterisk_var_run_t)
+	setattr_dirs_pattern($1, asterisk_var_run_t, asterisk_var_run_t)
+
+	files_search_pids($1)
+')
