diff -uNr refpolicy/policy/modules/kernel/kernel.te refpolicy/policy/modules/kernel/kernel.te
--- refpolicy/policy/modules/kernel/kernel.te	2012-07-26 21:10:04.774963777 +0200
+++ refpolicy/policy/modules/kernel/kernel.te	2012-07-26 21:10:06.766963782 +0200
@@ -254,6 +254,7 @@
 dev_create_generic_blk_files(kernel_t)
 dev_delete_generic_blk_files(kernel_t)
 dev_create_generic_chr_files(kernel_t)
+dev_setattr_generic_chr_files(kernel_t)
 dev_delete_generic_chr_files(kernel_t)
 dev_mounton(kernel_t)
 
