--- refpolicy/policy/modules/contrib/portage.te	2012-02-26 19:58:04.391566882 +0100
+++ refpolicy/policy/modules/contrib/portage.te	2012-02-26 19:56:58.584566742 +0100
@@ -12,6 +12,21 @@
 ## </desc>
 gen_tunable(portage_use_nfs, false)
 
+## <desc>
+## <p>
+## (deprecated) support for dontaudit tryouts
+## </p>
+## </desc>
+gen_tunable(gentoo_try_dontaudit, false)
+
+## <desc>
+## <p>
+## (deprecated) support for fixes
+## </p>
+## </desc>
+gen_tunable(gentoo_wait_requests, false)
+
+
 attribute_role portage_roles;
 
 type gcc_config_t;
