diff -uNr refpolicy/policy/modules/contrib/webalizer.te refpolicy/policy/modules/contrib/webalizer.te
--- refpolicy/policy/modules/contrib/webalizer.te	2012-07-26 21:09:59.582963767 +0200
+++ refpolicy/policy/modules/contrib/webalizer.te	2012-07-26 21:10:07.629963784 +0200
@@ -85,8 +85,10 @@
 userdom_use_unpriv_users_fds(webalizer_t)
 userdom_dontaudit_search_user_home_content(webalizer_t)
 
-apache_read_log(webalizer_t)
-apache_manage_sys_content(webalizer_t)
+optional_policy(`
+	apache_read_log(webalizer_t)
+	apache_manage_sys_content(webalizer_t)
+')
 
 optional_policy(`
 	cron_system_entry(webalizer_t, webalizer_exec_t)
