--- refpolicy/policy/modules/system/userdomain.if	2011-07-25 17:47:01.000000000 +0200
+++ refpolicy/policy/modules/system/userdomain.if	2011-10-09 15:25:03.501569994 +0200
@@ -243,6 +243,16 @@
 	filetrans_pattern($2, user_home_dir_t, user_home_t, { dir file lnk_file sock_file fifo_file })
 	files_list_home($2)
 
+	# manage user xdg locations
+	xdg_manage_generic_cache_home_content($2)
+	xdg_manage_generic_config_home_content($2)
+	xdg_manage_generic_data_home_content($2)
+	xdg_manage_generic_runtime_home_content($2)
+	xdg_relabel_generic_cache_home_content($2)
+	xdg_relabel_generic_config_home_content($2)
+	xdg_relabel_generic_data_home_content($2)
+	xdg_relabel_generic_runtime_home_content($2)
+
 	# cjp: this should probably be removed:
 	allow $2 user_home_dir_t:dir { manage_dir_perms relabel_dir_perms };
 
