diff -uNr refpolicy/policy/modules/roles/sysadm.te refpolicy/policy/modules/roles/sysadm.te
--- refpolicy/policy/modules/roles/sysadm.te	2012-07-26 21:10:04.305963777 +0200
+++ refpolicy/policy/modules/roles/sysadm.te	2012-07-26 21:10:08.034963784 +0200
@@ -152,6 +152,10 @@
 ')
 
 optional_policy(`
+	dracut_run(sysadm_t, sysadm_r)
+')
+
+optional_policy(`
 	firstboot_run(sysadm_t, sysadm_r)
 ')
 
