--- refpolicy/policy/modules/services/vde.te	2011-11-13 10:26:00.854873406 +0100
+++ refpolicy/policy/modules/services/vde.te	2011-11-13 10:24:50.873873258 +0100
@@ -12,9 +12,6 @@
 type vde_initrc_exec_t;
 init_script_file(vde_initrc_exec_t)
 
-type vde_conf_t;
-files_type(vde_conf_t);
-
 type vde_var_lib_t;
 files_type(vde_var_lib_t)
 
@@ -44,10 +41,6 @@
 
 files_read_etc_files(vde_t)
 
-allow vde_t vde_conf_t:dir list_dir_perms;
-read_files_pattern(vde_t, vde_conf_t, vde_conf_t)
-read_lnk_files_pattern(vde_t, vde_conf_t, vde_conf_t)
-
 domain_use_interactive_fds(vde_t)
 userdom_use_user_terminals(vde_t)
 miscfiles_read_localization(vde_t)
--- refpolicy/policy/modules/services/vde.if	2011-11-13 10:26:00.854873406 +0100
+++ refpolicy/policy/modules/services/vde.if	2011-11-13 10:24:40.499873234 +0100
@@ -19,7 +19,7 @@
 interface(`vde_role',`
 	gen_require(`
 		type vde_t, vde_tmp_t;
-		type vde_conf_t, vde_var_run_t;
+		type vde_var_run_t;
 		type vde_initrc_exec_t, vde_exec_t;
 	')
 
--- refpolicy/policy/modules/services/vde.fc	2011-11-13 10:26:00.854873406 +0100
+++ refpolicy/policy/modules/services/vde.fc	2011-11-13 10:24:35.047873224 +0100
@@ -1,7 +1,5 @@
 /etc/init.d/vde		--	gen_context(system_u:object_r:vde_initrc_exec_t,s0)
 /usr/bin/vde_switch	--	gen_context(system_u:object_r:vde_exec_t,s0)
 /usr/sbin/vde_tunctl	--	gen_context(system_u:object_r:vde_exec_t,s0)
-/etc/vde2(/.*)?			gen_context(system_u:object_r:vde_conf_t,s0)
-/etc/vde2/libvdemgmt(/.*)?	gen_context(system_u:object_r:vde_conf_t,s0)
 /var/run/vde\.ctl(/.*)?		gen_context(system_u:object_r:vde_var_run_t,s0)
 /tmp/vde.[0-9-]*	-s	gen_context(system_u:object_r:vde_tmp_t,s0)
