--- refpolicy/policy/modules/system/init.te	2012-03-09 22:30:09.714015394 +0100
+++ refpolicy/policy/modules/system/init.te	2012-03-09 22:30:04.719015383 +0100
@@ -119,6 +119,7 @@
 
 # Modify utmp.
 allow init_t initrc_var_run_t:file { rw_file_perms setattr };
+manage_dirs_pattern(initrc_t, initrc_var_run_t, initrc_var_run_t)
 
 kernel_read_system_state(init_t)
 kernel_share_state(init_t)
@@ -250,7 +251,7 @@
 manage_fifo_files_pattern(initrc_t, initrc_state_t, initrc_state_t)
 
 allow initrc_t initrc_var_run_t:file manage_file_perms;
-files_pid_filetrans(initrc_t, initrc_var_run_t, file)
+files_pid_filetrans(initrc_t, initrc_var_run_t, { file dir })
 
 can_exec(initrc_t, initrc_tmp_t)
 manage_files_pattern(initrc_t, initrc_tmp_t, initrc_tmp_t)
@@ -337,6 +338,7 @@
 files_getattr_all_symlinks(initrc_t)
 files_getattr_all_pipes(initrc_t)
 files_getattr_all_sockets(initrc_t)
+files_create_pid_dirs(initrc_t)
 files_purge_tmp(initrc_t)
 files_delete_all_locks(initrc_t)
 files_read_all_pids(initrc_t)
