--- refpolicy/policy/modules/system/init.te	2012-03-13 20:54:00.465460583 +0100
+++ refpolicy/policy/modules/system/init.te	2012-03-13 20:52:59.916460453 +0100
@@ -408,6 +408,7 @@
 logging_read_all_logs(initrc_t)
 logging_append_all_logs(initrc_t)
 logging_read_audit_config(initrc_t)
+logging_delete_devlog_socket(initrc_t)
 
 miscfiles_read_localization(initrc_t)
 # slapd needs to read cert files from its initscript
