--- refpolicy/policy/modules/services/inetd.te	2011-11-27 19:31:33.570009447 +0100
+++ refpolicy/policy/modules/services/inetd.te	2011-11-27 19:27:17.844008926 +0100
@@ -91,6 +91,7 @@
 corenet_udp_bind_inetd_child_port(inetd_t)
 corenet_tcp_bind_ircd_port(inetd_t)
 corenet_udp_bind_ktalkd_port(inetd_t)
+corenet_tcp_bind_pop_port(inetd_t)
 corenet_tcp_bind_printer_port(inetd_t)
 corenet_udp_bind_rlogind_port(inetd_t)
 corenet_udp_bind_rsh_port(inetd_t)
