--- refpolicy/policy/modules/system/sysnetwork.te	2012-03-25 16:43:39.886353485 +0200
+++ refpolicy/policy/modules/system/sysnetwork.te	2012-03-25 16:43:17.075353438 +0200
@@ -44,7 +44,7 @@
 # DHCP client local policy
 #
 allow dhcpc_t self:capability { dac_override fsetid net_admin net_raw net_bind_service setpcap sys_nice sys_resource sys_tty_config };
-dontaudit dhcpc_t self:capability { sys_tty_config sys_ptrace };
+dontaudit dhcpc_t self:capability { sys_tty_config sys_ptrace sys_admin };
 # for access("/etc/bashrc", X_OK) on Red Hat
 dontaudit dhcpc_t self:capability { dac_read_search sys_module };
 allow dhcpc_t self:process { getsched getcap setcap setfscreate ptrace signal_perms };
