--- refpolicy/policy/modules/contrib/nagios.te	2012-07-25 20:33:05.000000000 +0200
+++ refpolicy/policy/modules/contrib/nagios.te	2012-08-11 15:34:56.221495726 +0200
@@ -300,7 +300,7 @@
 
 optional_policy(`
 	postfix_stream_connect_master(nagios_mail_plugin_t)
-	posftix_exec_postqueue(nagios_mail_plugin_t)
+	postfix_exec_postqueue(nagios_mail_plugin_t)
 ')
 
 ######################################
