--- refpolicy/policy/modules/system/sysnetwork.if	2011-12-31 16:00:37.576751627 +0100
+++ refpolicy/policy/modules/system/sysnetwork.if	2011-12-31 15:58:40.330751377 +0100
@@ -83,6 +83,27 @@
 
 ########################################
 ## <summary>
+##	Do not audit attempts to read/write to
+##	the dhcp unix stream socket descriptors.
+## </summary>
+## <param name="domain">
+##	<summary>
+##	Domain to not audit.
+##	</summary>
+## </param>
+#
+interface(`sysnet_dontaudit_rw_dhcpc_unix_stream_sockets',`
+	gen_require(`
+		type dhcpc_t;
+	')
+
+	dontaudit $1 dhcpc_t:unix_stream_socket { read write };
+')
+
+
+
+########################################
+## <summary>
 ##	Send a SIGCHLD signal to the dhcp client.
 ## </summary>
 ## <param name="domain">
