--- refpolicy/policy/modules/contrib/djbdns.te	2011-10-25 19:19:59.000000000 +0200
+++ refpolicypolicy/modules/contrib/djbdns.te	2012-06-02 08:51:50.483113701 +0200
@@ -39,7 +39,9 @@
 
 files_search_var(djbdns_axfrdns_t)
 
-ucspitcp_service_domain(djbdns_axfrdns_t, djbdns_axfrdns_exec_t)
+optional_policy(`
+	ucspitcp_service_domain(djbdns_axfrdns_t, djbdns_axfrdns_exec_t)
+')
 
 ########################################
 #
