--- refpolicy/policy/modules/contrib/cvs.te	2011-10-25 19:19:59.000000000 +0200
+++ refpolicypolicy/modules/contrib/cvs.te	2012-06-02 09:07:11.906115669 +0200
@@ -14,7 +14,6 @@
 
 type cvs_t;
 type cvs_exec_t;
-inetd_tcp_service_domain(cvs_t, cvs_exec_t)
 application_executable_file(cvs_exec_t)
 role system_r types cvs_t;
 
@@ -96,6 +95,10 @@
 ')
 
 optional_policy(`
+	inetd_tcp_service_domain(cvs_t, cvs_exec_t)
+')
+
+optional_policy(`
 	kerberos_keytab_template(cvs, cvs_t)
 	kerberos_read_config(cvs_t)
 	kerberos_dontaudit_write_config(cvs_t)
