--- refpolicy/policy/modules/contrib/modemmanager.te	2011-10-25 19:20:02.000000000 +0200
+++ refpolicypolicy/modules/contrib/modemmanager.te	2012-06-08 20:57:05.618004661 +0200
@@ -34,7 +34,9 @@
 
 logging_send_syslog_msg(modemmanager_t)
 
-networkmanager_dbus_chat(modemmanager_t)
+optional_policy(`
+	networkmanager_dbus_chat(modemmanager_t)
+')
 
 optional_policy(`
 	udev_read_db(modemmanager_t)
