--- refpolicy/policy/modules/admin/portage.te	2012-01-21 20:50:57.973084118 +0100
+++ refpolicy/policy/modules/admin/portage.te	2012-01-21 20:49:35.359080808 +0100
@@ -260,6 +260,7 @@
 allow portage_fetch_t self:fifo_file rw_fifo_file_perms;
 allow portage_fetch_t self:tcp_socket create_stream_socket_perms;
 allow portage_fetch_t self:unix_stream_socket create_socket_perms;
+allow portage_fetch_t portage_devpts_t:chr_file { rw_chr_file_perms setattr };
 allow portage_fetch_t portage_tmp_t:dir manage_dir_perms;
 allow portage_fetch_t portage_tmp_t:file manage_file_perms;
 allow portage_fetch_t portage_conf_t:dir list_dir_perms;
