--- refpolicy/policy/modules/system/udev.te	2012-07-10 20:25:54.330224540 +0200
+++ refpolicypolicy/modules/system/udev.te	2012-07-10 19:50:11.729256326 +0200
@@ -174,6 +174,8 @@
 
 userdom_dontaudit_search_user_home_content(udev_t)
 
+udev_pid_filetrans_db(udev_t, dir, "data")
+
 ifdef(`distro_gentoo',`
 	# during boot, init scripts use /dev/.rcsysinit
 	# existance to determine if we are in early booting
--- refpolicy/policy/modules/system/udev.if	2012-07-10 20:25:54.330224540 +0200
+++ refpolicypolicy/modules/system/udev.if	2012-07-10 19:49:27.539256982 +0200
@@ -276,6 +276,56 @@
 
 ########################################
 ## <summary>
+##	Create udev database directories
+## </summary>
+## <param name="domain">
+##	<summary>
+##	Domain allowed access.
+##	</summary>
+## </param>
+#
+interface(`udev_create_db_dirs',`
+	gen_require(`
+		type udev_tbl_t;
+		type udev_var_run_t;
+	')
+
+	create_dirs_pattern($1, udev_var_run_t, udev_tbl_t)
+')
+
+
+
+########################################
+## <summary>
+##	Write in /var/run/udev with the udev_tbl_t (udev database) file type
+## </summary>
+## <param name="domain">
+##	<summary>
+##	Domain allowed access.
+##	</summary>
+## </param>
+## <param name="class">
+## 	<summary>
+##	Classes on which the file transition should occur
+##	</summary>
+## </param>
+## <param name="name" optional="true">
+##	<summary>
+##	Name of the directory that the file transition will work on
+##	</summary>
+## </param>
+#
+interface(`udev_pid_filetrans_db',`
+	gen_require(`
+		type udev_tbl_t;
+		type udev_var_run_t;
+	')
+
+	filetrans_pattern($1, udev_var_run_t, udev_tbl_t, $2, $3)
+')
+
+########################################
+## <summary>
 ##	Write dirs in /var/run with the udev_var_run file type
 ## </summary>
 ## <param name="domain">
@@ -289,7 +339,7 @@
 ##	</summary>
 ## </param>
 #
-interface(`udev_pid_filetrans_run_dirs',`
+interface(`udev_generic_pid_filetrans_run_dirs',`
 	gen_require(`
 		type udev_var_run_t;
 	')
--- refpolicy/policy/modules/system/udev.fc	2012-07-10 20:25:54.341224539 +0200
+++ refpolicypolicy/modules/system/udev.fc	2012-07-10 20:25:11.754225171 +0200
@@ -31,7 +31,7 @@
 /usr/bin/udevinfo --	gen_context(system_u:object_r:udev_exec_t,s0)
 
 /var/run/PackageKit/udev(/.*)? gen_context(system_u:object_r:udev_var_run_t,s0)
-/var/run/udev	-d	gen_context(system_u:object_r:udev_var_run_t,s0)
+/var/run/udev(/.*)?		gen_context(system_u:object_r:udev_var_run_t,s0)
 /var/run/udev/rules\.d(/.*)?	gen_context(system_u:object_r:udev_tbl_t,s0)
 /var/run/udev/data(/.*)?	gen_context(system_u:object_r:udev_tbl_t,s0)
 
