--- refpolicy/policy/modules/system/init.te	2012-07-17 19:02:54.107162338 +0200
+++ refpolicypolicy/modules/system/init.te	2012-07-17 19:02:30.249162691 +0200
@@ -641,8 +641,10 @@
 
 optional_policy(`
 	dbus_connect_system_bus(initrc_t)
+	dbus_create_system_dbusd_var_run_dirs(initrc_t)
 	dbus_system_bus_client(initrc_t)
 	dbus_read_config(initrc_t)
+	dbus_generic_pid_filetrans_system_dbusd_var_run(initrc_t, dir, "dbus")
 
 	optional_policy(`
 		consolekit_dbus_chat(initrc_t)
