--- refpolicy/policy/modules/kernel/devices.if	2012-07-25 12:20:10.246711498 +0200
+++ refpolicypolicy/modules/kernel/devices.if	2012-07-25 12:19:41.187711503 +0200
@@ -4001,6 +4001,26 @@
 
 ########################################
 ## <summary>
+##	Dont audit attempts to read hardware state information
+## </summary>
+## <param name="domain">
+##	<summary>
+##	Domain for which the attempts do not need to be audited
+##	</summary>
+## </param>
+#
+interface(`dev_dontaudit_read_sysfs',`
+	gen_require(`
+		type sysfs_t;
+	')
+
+	dontaudit $1 sysfs_t:file read_file_perms;
+	dontaudit $1 sysfs_t:dir list_dir_perms;
+	dontaudit $1 sysfs_t:lnk_file read_lnk_file_perms;
+')
+
+########################################
+## <summary>
 ##	Read hardware state information.
 ## </summary>
 ## <desc>
