--- refpolicy/policy/modules/contrib/java.te	2012-02-15 20:32:44.000000000 +0100
+++ refpolicypolicy/modules/contrib/java.te	2012-07-23 19:20:28.807396492 +0200
@@ -39,6 +39,7 @@
 
 allow java_t self:process { signal_perms getsched setsched execmem };
 allow java_t self:fifo_file rw_fifo_file_perms;
+allow java_t self:sem create_sem_perms;
 allow java_t self:tcp_socket create_socket_perms;
 allow java_t self:udp_socket create_socket_perms;
 
@@ -97,7 +98,7 @@
 
 sysnet_read_config(java_t)
 
-userdom_dontaudit_use_user_terminals(java_t)
+userdom_use_user_terminals(java_t)
 userdom_dontaudit_setattr_user_home_content_files(java_t)
 userdom_dontaudit_exec_user_home_content_files(java_t)
 userdom_manage_user_home_content_dirs(java_t)
@@ -120,6 +121,20 @@
 ')
 
 optional_policy(`
+	alsa_read_rw_config(java_t)
+')
+
+optional_policy(`
+	# Plugin communication
+	chromium_rw_tmp_pipes(java_t)
+')
+
+optional_policy(`
+	# Plugin communication
+	mozilla_rw_tmp_pipes(java_t)
+')
+
+optional_policy(`
 	nis_use_ypbind(java_t)
 ')
 
--- refpolicy/policy/modules/contrib/java.if	2011-10-25 19:20:01.000000000 +0200
+++ refpolicypolicy/modules/contrib/java.if	2012-07-23 21:25:05.715412450 +0200
@@ -115,6 +115,32 @@
 
 ########################################
 ## <summary>
+##	Run java in javaplugin domain and
+##	do not clean the environment (atsecure)
+## </summary>
+## <desc>
+##	<p>
+##	This is needed when java is called by an application with library
+##	settings (such as is the case when invoked as a browser plugin)
+##	</p>
+## </desc>
+## <param name="domain">
+##	<summary>
+##	Domain allowed to transition.
+##	</summary>
+## </param>
+#
+template(`java_noatsecure_domtrans',`
+	gen_require(`
+		type java_t;
+	')
+
+	allow $1 java_t:process noatsecure;
+
+	java_domtrans($1)
+')
+########################################
+## <summary>
 ##	Execute java in the java domain, and
 ##	allow the specified role the java domain.
 ## </summary>
