--- refpolicy/policy/modules/kernel/kernel.if	2011-10-12 20:08:44.000000000 +0200
+++ refpolicypolicy/modules/kernel/kernel.if	2012-07-23 14:42:54.253360974 +0200
@@ -1843,6 +1843,24 @@
 	list_dirs_pattern($1, { proc_t sysctl_t }, sysctl_crypto_t)
 ')
 
+#######################################
+## <summary>
+##	Do not audit attempted reading of kernel sysctls
+## </summary>
+## <param name="domain">
+##	<summary>
+##	Domain to not audit accesses from
+##	</summary>
+## </param>
+#
+interface(`kernel_dontaudit_read_kernel_sysctls',`
+	gen_require(`
+		type sysctl_kernel_t;
+	')
+
+	dontaudit $1 sysctl_kernel_t:file read_file_perms;
+')
+
 ########################################
 ## <summary>
 ##	Read general kernel sysctls.
