--- refpolicy/policy/modules/contrib/policykit.te	2012-05-04 15:15:21.000000000 +0200
+++ refpolicy/policy/modules/contrib/policykit.te	2012-09-08 19:21:10.406561687 +0200
@@ -1,4 +1,4 @@
-policy_module(policykit, 1.2.0)
+policy_module(policykit, 1.2.1)
 
 ########################################
 #
--- refpolicy/policy/modules/contrib/policykit.fc	2012-05-04 15:15:21.000000000 +0200
+++ refpolicy/policy/modules/contrib/policykit.fc	2012-09-08 19:21:03.900561680 +0200
@@ -1,3 +1,6 @@
+/usr/lib/polkit-1/polkitd		--	gen_context(system_u:object_r:policykit_exec_t,s0)
+/usr/lib/polkit-1/polkit-agent-helper-1	--	gen_context(system_u:object_r:policykit_auth_exec_t,s0)
+
 /usr/lib/policykit/polkit-read-auth-helper --	gen_context(system_u:object_r:policykit_auth_exec_t,s0)
 /usr/lib/policykit/polkit-grant-helper.* --	gen_context(system_u:object_r:policykit_grant_exec_t,s0)
 /usr/lib/policykit/polkit-resolve-exe-helper.* -- gen_context(system_u:object_r:policykit_resolve_exec_t,s0)
