--- refpolicy-20110726/policy/modules/services/rpc.te	2011-07-26 14:10:40.000000000 +0200
+++ refpolicy/policy/modules/services/rpc.te	2011-07-27 07:42:41.555004759 +0200
@@ -61,6 +61,7 @@
 allow rpcd_t self:capability { sys_admin chown dac_override setgid setuid };
 allow rpcd_t self:process { getcap setcap };
 allow rpcd_t self:fifo_file rw_fifo_file_perms;
+allow rpcd_t self:udp_socket { listen };
 
 allow rpcd_t rpcd_var_run_t:dir setattr;
 manage_files_pattern(rpcd_t, rpcd_var_run_t, rpcd_var_run_t)
