--- refpolicy.orig/policy/modules/services/courier.te	2010-12-13 15:11:02.000000000 +0100
+++ refpolicy/policy/modules/services/courier.te	2011-03-13 12:29:52.793000000 +0100
@@ -37,7 +37,7 @@
 #
 
 allow courier_authdaemon_t self:capability { setuid setgid sys_tty_config };
-allow courier_authdaemon_t self:unix_stream_socket connectto;
+allow courier_authdaemon_t self:unix_stream_socket {  create_stream_socket_perms connectto };
 
 can_exec(courier_authdaemon_t, courier_exec_t)
 
@@ -52,7 +52,9 @@
 allow courier_authdaemon_t courier_tcpd_t:tcp_socket rw_stream_socket_perms;
 allow courier_authdaemon_t courier_tcpd_t:fifo_file rw_file_perms;
 
+manage_dirs_pattern(courier_authdaemon_t, courier_var_lib_t, courier_var_lib_t)
 manage_sock_files_pattern(courier_authdaemon_t, courier_spool_t, courier_spool_t)
+manage_sock_files_pattern(courier_authdaemon_t, courier_var_lib_t, courier_var_lib_t)
 files_search_spool(courier_authdaemon_t)
 
 corecmd_search_bin(courier_authdaemon_t)
@@ -133,6 +135,7 @@
 miscfiles_read_localization(courier_tcpd_t)
 
 courier_domtrans_pop(courier_tcpd_t)
+courier_authdaemon_stream_connect(courier_tcpd_t)
 
 ########################################
 #
--- refpolicy.orig/policy/modules/services/courier.if	2010-09-15 15:14:53.000000000 +0200
+++ refpolicy/policy/modules/services/courier.if	2011-03-13 12:29:18.089999999 +0100
@@ -106,6 +106,25 @@
 
 ########################################
 ## <summary>
+##	Connect to Authdaemon using a unix domain stream socket.
+## </summary>
+## <param name="domain">
+##	<summary>
+##	Domain allowed access.
+##	</summary>
+## </param>
+## <rolecap/>
+#
+interface(`courier_authdaemon_stream_connect',`
+	gen_require(`
+		type courier_authdaemon_t, courier_var_run_t;
+	')
+
+	stream_connect_pattern($1, courier_var_run_t, courier_var_run_t, courier_authdaemon_t)
+')
+
+########################################
+## <summary>
 ##	Execute the courier POP3 and IMAP server with
 ##	a domain transition.
 ## </summary>
--- refpolicy.orig/policy/modules/services/courier.fc	2010-08-03 15:11:05.000000000 +0200
+++ refpolicy/policy/modules/services/courier.fc	2011-03-13 12:15:07.430999999 +0100
@@ -5,8 +5,10 @@
 /usr/sbin/courierlogger			--	gen_context(system_u:object_r:courier_exec_t,s0)
 /usr/sbin/courierldapaliasd		--	gen_context(system_u:object_r:courier_exec_t,s0)
 /usr/sbin/couriertcpd			--	gen_context(system_u:object_r:courier_tcpd_exec_t,s0)
-
-/usr/lib(64)?/courier/authlib/.*	--	gen_context(system_u:object_r:courier_authdaemon_exec_t,s0)
+ifdef(`distro_gentoo',`
+/usr/lib(64)?/courier-imap/couriertcpd	--	gen_context(system_u:object_r:courier_tcpd_exec_t,s0)
+')
+/usr/lib(64)?/courier/(courier-)?authlib/.*	--	gen_context(system_u:object_r:courier_authdaemon_exec_t,s0)
 /usr/lib(64)?/courier/courier/.*	--	gen_context(system_u:object_r:courier_exec_t,s0)
 /usr/lib(64)?/courier/courier/courierpop.* --	gen_context(system_u:object_r:courier_pop_exec_t,s0)
 /usr/lib(64)?/courier/courier/imaplogin --	gen_context(system_u:object_r:courier_pop_exec_t,s0)
@@ -16,9 +18,13 @@
 /usr/lib(64)?/courier/rootcerts(/.*)?		gen_context(system_u:object_r:courier_etc_t,s0)
 /usr/lib(64)?/courier/sqwebmail/cleancache\.pl -- gen_context(system_u:object_r:sqwebmail_cron_exec_t,s0)
 
-/var/lib/courier(/.*)?			--	gen_context(system_u:object_r:courier_var_lib_t,s0)
+/var/lib/courier(/.*)?				gen_context(system_u:object_r:courier_var_lib_t,s0)
 
-/var/run/courier(/.*)?			--	gen_context(system_u:object_r:courier_var_run_t,s0)
+/var/run/courier(/.*)?				gen_context(system_u:object_r:courier_var_run_t,s0)
+# Courier authdaemon config file has this location, but sais not to touch it!
+ifdef(`distro_gentoo',`
+/var/lib/courier/authdaemon(/.*)?		gen_context(system_u:object_r:courier_var_run_t,s)
+')
 
 /var/spool/authdaemon(/.*)?			gen_context(system_u:object_r:courier_spool_t,s0)
 /var/spool/courier(/.*)?			gen_context(system_u:object_r:courier_spool_t,s0)
