--- refpolicy.orig/policy/modules/system/init.te	2010-12-13 15:11:02.000000000 +0100
+++ refpolicy/policy/modules/system/init.te	2011-05-13 23:37:46.603000085 +0200
@@ -70,6 +70,7 @@
 
 type initrc_state_t;
 files_type(initrc_state_t)
+files_mountpoint(initrc_state_t)
 
 type initrc_tmp_t;
 files_tmp_file(initrc_tmp_t)
@@ -169,6 +170,9 @@
 miscfiles_read_localization(init_t)
 
 ifdef(`distro_gentoo',`
+	optional_policy(`
+		alsa_read_lib(initrc_t)
+	')
 	allow init_t self:process { getcap setcap };
 ')
 
@@ -844,3 +848,7 @@
 optional_policy(`
 	zebra_read_config(initrc_t)
 ')
+
+optional_policy(`
+	courier_read_config(initrc_t)
+')
--- refpolicy.orig/policy/modules/system/init.if	2010-10-11 15:32:08.000000000 +0200
+++ refpolicy/policy/modules/system/init.if	2011-05-13 21:02:37.064000084 +0200
@@ -756,6 +756,11 @@
 	gen_require(`
 		type initrc_exec_t;
 	')
+	
+	# /sbin/runscript is a wrapper for /sbin/rc, so run_init_t
+	# wants to execute initrc_exec_t (no transition needed anymore) whereas
+	# runscript previously was a binary
+	allow $1 initrc_exec_t:file execute_no_trans;
 
 	domain_entry_file($1, initrc_exec_t)
 ')
--- refpolicy.orig/policy/modules/system/init.fc	2010-08-03 15:11:04.000000000 +0200
+++ refpolicy/policy/modules/system/init.fc	2011-05-13 20:30:26.136000088 +0200
@@ -51,6 +51,7 @@
 ifdef(`distro_gentoo', `
 /var/lib/init\.d(/.*)?		gen_context(system_u:object_r:initrc_state_t,s0)
 /var/run/svscan\.pid	--	gen_context(system_u:object_r:initrc_var_run_t,s0)
+/lib(64)?/rc/init\.d(/.*)?		gen_context(system_u:object_r:initrc_state_t,s0)
 ')
 
 /var/run/utmp		--	gen_context(system_u:object_r:initrc_var_run_t,s0)
