--- refpolicy.orig/policy/modules/system/iptables.if	2010-08-05 15:25:55.000000000 +0200
+++ refpolicy/policy/modules/system/iptables.if	2011-01-31 20:22:49.257000083 +0100
@@ -165,3 +165,21 @@
 	files_search_etc($1)
 	manage_files_pattern($1, iptables_conf_t, iptables_conf_t)
 ')
+
+###################################
+## <summary>
+##   Do not audit rw rawip_socket operations
+## </summary>
+## <param name="domain">
+##   <summary>
+##     Domain to ignore
+##   </summary>
+## </param>
+#
+interface(`gentoo_iptables_dontaudit_rw_rawip_socket',`
+	gen_require(`
+		type iptables_t;
+	')
+
+	dontaudit $1 iptables_t:rawip_socket { read write };
+')
