--- refpolicy-20110726/policy/modules/services/puppet.te	2011-07-26 14:10:40.000000000 +0200
+++ refpolicy/policy/modules/services/puppet.te	2011-07-27 18:24:27.843005853 +0200
@@ -50,7 +50,7 @@
 # Puppet personal policy
 #
 
-allow puppet_t self:capability { fowner fsetid setuid setgid dac_override sys_nice sys_ptrace sys_tty_config };
+allow puppet_t self:capability { fowner fsetid setuid setgid dac_override sys_nice sys_ptrace sys_tty_config chown };
 allow puppet_t self:process { signal signull getsched setsched };
 allow puppet_t self:fifo_file rw_fifo_file_perms;
 allow puppet_t self:netlink_route_socket create_netlink_socket_perms;
@@ -77,7 +77,8 @@
 files_tmp_filetrans(puppet_t, puppet_tmp_t, { file dir })
 
 kernel_dontaudit_search_sysctl(puppet_t)
-kernel_dontaudit_search_kernel_sysctl(puppet_t)
+kernel_read_kernel_sysctls(puppet_t)
+kernel_read_network_state(puppet_t)
 kernel_read_system_state(puppet_t)
 kernel_read_crypto_sysctls(puppet_t)
 
@@ -131,6 +132,13 @@
 sysnet_dns_name_resolve(puppet_t)
 sysnet_run_ifconfig(puppet_t, system_r)
 
+usermanage_domtrans_passwd(puppet_t)
+
+tunable_policy(`gentoo_try_dontaudit',`
+	dontaudit puppet_t self:capability dac_read_search;
+	userdom_dontaudit_use_user_terminals(puppet_t)
+')
+
 tunable_policy(`puppet_manage_all_files',`
 	auth_manage_all_files_except_auth_files(puppet_t)
 ')
@@ -144,6 +152,15 @@
 ')
 
 optional_policy(`
+	mta_send_mail(puppet_t)
+')
+
+optional_policy(`
+	gentoo_init_rc_exec(puppet_t)
+	portage_run(puppet_t, system_r)
+')
+
+optional_policy(`
 	files_rw_var_files(puppet_t)
 
 	rpm_domtrans(puppet_t)
--- refpolicy-20110726/policy/modules/services/puppet.fc	2010-08-03 15:11:07.000000000 +0200
+++ refpolicy/policy/modules/services/puppet.fc	2011-07-27 18:25:00.571005854 +0200
@@ -3,7 +3,9 @@
 /etc/rc\.d/init\.d/puppet	--	gen_context(system_u:object_r:puppet_initrc_exec_t,s0)
 /etc/rc\.d/init\.d/puppetmaster --	gen_context(system_u:object_r:puppetmaster_initrc_exec_t,s0)
 
+/usr/bin/puppetd		--	gen_context(system_u:object_r:puppet_exec_t,s0)
 /usr/sbin/puppetd		--	gen_context(system_u:object_r:puppet_exec_t,s0)
+/usr/bin/puppetmasterd		--	gen_context(system_u:object_r:puppetmaster_exec_t,s0)
 /usr/sbin/puppetmasterd		--	gen_context(system_u:object_r:puppetmaster_exec_t,s0)
 
 /var/lib/puppet(/.*)?			gen_context(system_u:object_r:puppet_var_lib_t,s0)
