--- refpolicy.orig/policy/modules/system/init.te	2010-12-13 15:11:02.000000000 +0100
+++ refpolicy/policy/modules/system/init.te	2011-07-21 11:14:54.520000370 +0200
@@ -56,8 +56,10 @@
 
 type initrc_t, init_script_domain_type, init_run_all_scripts_domain;
 type initrc_exec_t, init_script_file_type;
+type rc_exec_t;
 domain_type(initrc_t)
 domain_entry_file(initrc_t, initrc_exec_t)
+domain_entry_file(initrc_t, rc_exec_t)
 role system_r types initrc_t;
 # should be part of the true block
 # of the below init_upstart tunable
@@ -70,6 +72,7 @@
 
 type initrc_state_t;
 files_type(initrc_state_t)
+files_mountpoint(initrc_state_t)
 
 type initrc_tmp_t;
 files_tmp_file(initrc_tmp_t)
@@ -169,6 +172,9 @@
 miscfiles_read_localization(init_t)
 
 ifdef(`distro_gentoo',`
+	optional_policy(`
+		alsa_read_lib(initrc_t)
+	')
 	allow init_t self:process { getcap setcap };
 ')
 
@@ -233,6 +239,7 @@
 can_exec(initrc_t, init_script_file_type)
 
 domtrans_pattern(init_run_all_scripts_domain, initrc_exec_t, initrc_t)
+gentoo_init_rc_exec(initrc_t)
 
 manage_dirs_pattern(initrc_t, initrc_state_t, initrc_state_t)
 manage_files_pattern(initrc_t, initrc_state_t, initrc_state_t)
@@ -783,6 +790,10 @@
 optional_policy(`
 	# allow init scripts to su
 	su_restricted_domain_template(initrc, initrc_t, system_r)
+	# Allow initrc_su_t, now defined, to transition to postgresql_t
+	postgresql_domtrans(initrc_su_t)
+	# Allow initrc_su_t to use the initrc_devpts_t (needed for init script failure output)
+	allow initrc_su_t initrc_devpts_t:chr_file { read write };
 ')
 
 optional_policy(`
@@ -844,3 +855,7 @@
 optional_policy(`
 	zebra_read_config(initrc_t)
 ')
+
+optional_policy(`
+	courier_read_config(initrc_t)
+')
--- refpolicy.orig/policy/modules/system/init.if	2010-10-11 15:32:08.000000000 +0200
+++ refpolicy/policy/modules/system/init.if	2011-07-21 11:17:13.913000374 +0200
@@ -454,6 +454,29 @@
 
 ########################################
 ## <summary>
+## Execute the rc program in the caller domain. This is a fix for OpenRC,
+## used in distributions like Gentoo, where /sbin/rc is used for all kinds
+## of stuff and as such cannot be handled properly using initrc_exec_t
+## </summary>
+## <param name="domain">
+##	<summary>
+##	Domain allowed access.
+##	</summary>
+## </param>
+## <rolecap/>
+#
+interface(`gentoo_init_rc_exec',`
+	gen_require(`
+		type rc_exec_t;
+	')
+
+	corecmd_search_bin($1)
+	can_exec($1, rc_exec_t)
+')
+
+
+########################################
+## <summary>
 ##	Get the process group of init.
 ## </summary>
 ## <param name="domain">
@@ -756,6 +779,11 @@
 	gen_require(`
 		type initrc_exec_t;
 	')
+	
+	# /sbin/runscript is a wrapper for /sbin/rc, so run_init_t
+	# wants to execute initrc_exec_t (no transition needed anymore) whereas
+	# runscript previously was a binary
+	allow $1 initrc_exec_t:file execute_no_trans;
 
 	domain_entry_file($1, initrc_exec_t)
 ')
@@ -799,11 +827,12 @@
 #
 interface(`init_domtrans_script',`
 	gen_require(`
-		type initrc_t, initrc_exec_t;
+		type initrc_t, initrc_exec_t, rc_exec_t;
 	')
 
 	files_list_etc($1)
 	domtrans_pattern($1, initrc_exec_t, initrc_t)
+	domtrans_pattern($1, rc_exec_t, initrc_t)
 
 	ifdef(`enable_mcs',`
 		range_transition $1 initrc_exec_t:process s0;
--- refpolicy.orig/policy/modules/system/init.fc	2010-08-03 15:11:04.000000000 +0200
+++ refpolicy/policy/modules/system/init.fc	2011-07-21 14:42:51.031000726 +0200
@@ -27,7 +27,7 @@
 /sbin/init(ng)?		--	gen_context(system_u:object_r:init_exec_t,s0)
 
 ifdef(`distro_gentoo', `
-/sbin/rc		--	gen_context(system_u:object_r:initrc_exec_t,s0)
+/sbin/rc		--	gen_context(system_u:object_r:rc_exec_t,s0)
 /sbin/runscript		--	gen_context(system_u:object_r:initrc_exec_t,s0)
 /sbin/runscript\.sh	--	gen_context(system_u:object_r:initrc_exec_t,s0)
 /sbin/runsvcscript\.sh	--	gen_context(system_u:object_r:initrc_exec_t,s0)
@@ -50,7 +50,10 @@
 #
 ifdef(`distro_gentoo', `
 /var/lib/init\.d(/.*)?		gen_context(system_u:object_r:initrc_state_t,s0)
+/var/lib/iptables(/.*)?		gen_context(system_u:object_r:initrc_tmp_t,s0)
 /var/run/svscan\.pid	--	gen_context(system_u:object_r:initrc_var_run_t,s0)
+/lib64/rc/init\.d(/.*)?		gen_context(system_u:object_r:initrc_state_t,s0)
+/lib/rc/init\.d(/.*)?		gen_context(system_u:object_r:initrc_state_t,s0)
 ')
 
 /var/run/utmp		--	gen_context(system_u:object_r:initrc_var_run_t,s0)
