--- refpolicy-20110726/policy/modules/services/courier.te	2011-07-26 14:10:40.000000000 +0200
+++ refpolicy/policy/modules/services/courier.te	2011-07-26 23:36:49.220003887 +0200
@@ -52,6 +52,8 @@
 allow courier_authdaemon_t courier_tcpd_t:tcp_socket rw_stream_socket_perms;
 allow courier_authdaemon_t courier_tcpd_t:fifo_file rw_file_perms;
 
+read_lnk_files_pattern(courier_authdaemon_t, courier_var_lib_t, courier_var_lib_t)
+
 create_dirs_pattern(courier_authdaemon_t, courier_var_lib_t, courier_var_lib_t)
 manage_sock_files_pattern(courier_authdaemon_t, courier_spool_t, courier_spool_t)
 manage_sock_files_pattern(courier_authdaemon_t, courier_var_lib_t, courier_var_lib_t)
@@ -97,8 +99,13 @@
 # inherits file handle - should it?
 allow courier_pop_t courier_var_lib_t:file { read write };
 
+# TODO Correct this, mentioning "var_lib_t" here is not done.
+search_dirs_pattern(courier_pop_t, var_lib_t, courier_var_lib_t)
+read_lnk_files_pattern(courier_pop_t, var_lib_t, courier_var_lib_t)
+
 miscfiles_read_localization(courier_pop_t)
 
+courier_authdaemon_rw_inherited_stream_sockets(courier_pop_t)
 courier_domtrans_authdaemon(courier_pop_t)
 
 # do the actual work (read the Maildir)
@@ -135,6 +142,8 @@
 miscfiles_read_localization(courier_tcpd_t)
 
 courier_domtrans_pop(courier_tcpd_t)
+courier_authdaemon_stream_connect(courier_tcpd_t)
+courier_domtrans_authdaemon(courier_tcpd_t)
 
 ########################################
 #
@@ -146,3 +155,7 @@
 optional_policy(`
 	cron_system_entry(courier_sqwebmail_t, courier_sqwebmail_exec_t)
 ')
+
+optional_policy(`
+	mysql_stream_connect(courier_authdaemon_t)
+')
--- refpolicy-20110726/policy/modules/services/courier.if	2011-03-28 17:05:14.000000000 +0200
+++ refpolicy/policy/modules/services/courier.if	2011-07-26 23:38:41.329003890 +0200
@@ -49,6 +49,7 @@
 	kernel_read_kernel_sysctls(courier_$1_t)
 
 	corecmd_exec_bin(courier_$1_t)
+	corecmd_exec_shell(courier_$1_t)
 
 	corenet_all_recvfrom_unlabeled(courier_$1_t)
 	corenet_all_recvfrom_netlabel(courier_$1_t)
@@ -105,6 +106,44 @@
 ')
 
 ########################################
+## <summary>
+##     Allow read/write operations on an inherited stream socket
+## </summary>
+## <param name="domain">
+##     <summary>
+##     Domain allowed access.
+##     </summary>
+## </param>
+## <rolecap/>
+#
+interface(`courier_authdaemon_rw_inherited_stream_sockets',`
+       gen_require(`
+               type courier_authdaemon_t;
+       ')
+       allow $1 courier_authdaemon_t:unix_stream_socket { read write };
+')
+
+
+########################################
+## <summary>
+##     Connect to Authdaemon using a unix domain stream socket.
+## </summary>
+## <param name="domain">
+##     <summary>
+##     Domain allowed access.
+##     </summary>
+## </param>
+## <rolecap/>
+#
+interface(`courier_authdaemon_stream_connect',`
+       gen_require(`
+               type courier_authdaemon_t, courier_var_run_t;
+       ')
+
+       stream_connect_pattern($1, courier_var_run_t, courier_var_run_t, courier_authdaemon_t)
+')
+
+########################################
 ## <summary>
 ##	Execute the courier POP3 and IMAP server with
 ##	a domain transition.
--- refpolicy-20110726/policy/modules/services/courier.fc	2011-03-28 17:05:14.000000000 +0200
+++ refpolicy/policy/modules/services/courier.fc	2011-07-26 23:40:27.012003893 +0200
@@ -2,9 +2,11 @@
 
 /usr/bin/imapd				--	gen_context(system_u:object_r:courier_pop_exec_t,s0)
 
+/usr/sbin/courier-imapd			--	gen_context(system_u:object_r:courier_pop_exec_t,s0)
 /usr/sbin/courierlogger			--	gen_context(system_u:object_r:courier_exec_t,s0)
 /usr/sbin/courierldapaliasd		--	gen_context(system_u:object_r:courier_exec_t,s0)
 /usr/sbin/couriertcpd			--	gen_context(system_u:object_r:courier_tcpd_exec_t,s0)
+/usr/sbin/imaplogin			--	gen_context(system_u:object_r:courier_authdaemon_exec_t,s0)
 
 /usr/lib(64)?/courier/(courier-)?authlib/.*	--	gen_context(system_u:object_r:courier_authdaemon_exec_t,s0)
 /usr/lib(64)?/courier/courier/.*	--	gen_context(system_u:object_r:courier_exec_t,s0)
