--- refpolicy-20110726/policy/modules/system/iptables.if	2011-04-18 17:54:20.000000000 +0200
+++ refpolicy/policy/modules/system/iptables.if	2011-07-27 08:09:12.720005145 +0200
@@ -169,3 +169,22 @@
 	files_search_etc($1)
 	manage_files_pattern($1, iptables_conf_t, iptables_conf_t)
 ')
+
+###################################
+## <summary>
+##   Do not audit rw rawip_socket operations
+## </summary>
+## <param name="domain">
+##   <summary>
+##     Domain to ignore
+##   </summary>
+## </param>
+#
+interface(`gentoo_iptables_dontaudit_rw_rawip_socket',`
+       gen_require(`
+               type iptables_t;
+       ')
+
+       dontaudit $1 iptables_t:rawip_socket { read write };
+')
+
