--- refpolicy.orig/policy/modules/system/selinuxutil.te	2010-08-03 15:11:04.000000000 +0200
+++ refpolicy/policy/modules/system/selinuxutil.te	2011-07-22 19:55:07.676001201 +0200
@@ -402,6 +402,7 @@
 	ifdef(`distro_gentoo',`
 		# Gentoo integrated run_init:
 		init_script_file_entry_type(run_init_t)
+		gentoo_init_rc_exec(run_init_t)
 	')
 ')
 
@@ -498,6 +499,10 @@
 	userdom_read_user_tmp_files(semanage_t)
 ')
 
+optional_policy(`
+	gentoo_portage_eselect_module(semanage_t)
+')
+
 ########################################
 #
 # Setfiles local policy
@@ -579,6 +584,12 @@
 	fs_rw_tmpfs_chr_files(setfiles_t)
 ')
 
+ifdef(`distro_gentoo',`
+	tunable_policy(`gentoo_try_dontaudit',`
+		gentoo_portage_dontaudit_use_fds(load_policy_t)
+	')
+')
+
 ifdef(`distro_redhat', `
 	fs_rw_tmpfs_chr_files(setfiles_t)
 	fs_rw_tmpfs_blk_files(setfiles_t)
