--- refpolicy.orig/policy/modules/roles/staff.te	2010-12-13 15:11:01.000000000 +0100
+++ refpolicy/policy/modules/roles/staff.te	2011-01-22 15:17:47.821000048 +0100
@@ -171,4 +171,20 @@
 	optional_policy(`
 		wireshark_role(staff_r, staff_t)
 	')
+
+	optional_policy(`
+		gorg_role(staff_r, staff_t)
+	')
+
+	optional_policy(`
+		skype_role(staff_r, staff_t)
+	')
+
+	optional_policy(`
+		links_role(staff_r, staff_t)
+	')
+
+	optional_policy(`
+		mutt_role(staff_r, staff_t)
+	')
 ')
