--- refpolicy.orig/policy/modules/system/mount.te	2010-12-13 15:11:02.000000000 +0100
+++ refpolicy/policy/modules/system/mount.te	2011-04-04 22:12:41.978000000 +0200
@@ -48,6 +48,7 @@
 
 kernel_read_system_state(mount_t)
 kernel_read_kernel_sysctls(mount_t)
+kernel_setsched(mount_t)
 kernel_dontaudit_getattr_core_if(mount_t)
 kernel_dontaudit_write_debugfs_dirs(mount_t)
 kernel_dontaudit_write_proc_dirs(mount_t)
@@ -84,7 +85,7 @@
 # For reading cert files
 files_read_usr_files(mount_t)
 files_list_mnt(mount_t)
-files_dontaudit_write_root_dirs(mount_t)
+gentoo_files_dontaudit_write_all_mountpoints(mount_t)
 
 fs_getattr_xattr_fs(mount_t)
 fs_getattr_cifs(mount_t)
@@ -139,6 +140,12 @@
 	')
 ')
 
+ifdef(`distro_gentoo',`
+	tunable_policy(`gentoo_try_dontaudit',`
+		term_dontaudit_manage_pty_dirs(mount_t)
+	')
+')
+
 tunable_policy(`allow_mount_anyfile',`
 	auth_read_all_dirs_except_shadow(mount_t)
 	auth_read_all_files_except_shadow(mount_t)
