--- refpolicy.orig/policy/modules/kernel/corenetwork.te	2010-12-13 15:13:08.000000000 +0100
+++ refpolicy/policy/modules/kernel/corenetwork.te	2011-06-13 10:50:00.937000248 +0200
@@ -1449,7 +1449,6 @@
 type xserver_server_packet_t, packet_type, server_packet_type;
 portcon tcp 6000-6020 gen_context(system_u:object_r:xserver_port_t,s0)
 
-
 type zookeeper_client_port_t, port_type;
 type zookeeper_client_client_packet_t, packet_type, client_packet_type;
 type zookeeper_client_server_packet_t, packet_type, server_packet_type;
--- refpolicy.orig/policy/modules/kernel/corenetwork.if	2010-12-13 15:13:08.000000000 +0100
+++ refpolicy/policy/modules/kernel/corenetwork.if	2011-06-13 10:50:29.983000248 +0200
@@ -67355,9 +67355,6 @@
 	allow $1 xserver_server_packet_t:packet relabelto;
 ')
 
-
-
-
 ########################################
 ## <summary>
 ##	Send and receive TCP traffic on the zookeeper_client port.
@@ -67544,7 +67541,6 @@
 	allow $1 zookeeper_client_port_t:tcp_socket name_connect;
 ')
 
-
 ########################################
 ## <summary>
 ##	Send zookeeper_client_client packets.
--- refpolicy.orig/policy/modules/kernel/corenetwork.te.in	2010-12-13 15:11:03.000000000 +0100
+++ refpolicy/policy/modules/kernel/corenetwork.te.in	2011-06-13 10:10:57.693000180 +0200
@@ -213,6 +213,8 @@
 network_port(xen, tcp,8002,s0)
 network_port(xfs, tcp,7100,s0)
 network_port(xserver, tcp,6000-6020,s0)
+network_port(zabbix_agent, tcp,10050,s0)
+network_port(zabbix, tcp,10051,s0)
 network_port(zookeeper_client, tcp,2181,s0)
 network_port(zookeeper_election, tcp,3888,s0)
 network_port(zookeeper_leader, tcp,2888,s0)
