--- refpolicy.orig/policy/modules/roles/sysadm.te	2010-12-13 15:11:01.000000000 +0100
+++ refpolicy/policy/modules/roles/sysadm.te	2011-04-04 21:37:32.590000002 +0200
@@ -174,6 +174,10 @@
 ')
 
 optional_policy(`
+	ldap_admin(sysadm_t, sysadm_r)
+')
+
+optional_policy(`
 	libs_run_ldconfig(sysadm_t, sysadm_r)
 ')
 
@@ -261,7 +265,7 @@
 ')
 
 optional_policy(`
-	raid_domtrans_mdadm(sysadm_t)
+	gentoo_raid_mdadm_role(sysadm_r, sysadm_t)
 ')
 
 optional_policy(`
@@ -400,6 +404,22 @@
 	yam_run(sysadm_t, sysadm_r)
 ')
 
+optional_policy(`
+	networkmanager_run_wpa_cli(sysadm_t, sysadm_r)
+')
+
+optional_policy(`
+	postfix_run(master, sysadm_t, sysadm_r)
+	postfix_run(postqueue, sysadm_t, sysadm_r)
+')
+
+ifdef(`distro_gentoo', `
+	optional_policy(`
+		postgresql_admin(sysadm_t, sysadm_r)
+		postgresql_domtrans(sysadm_t)
+	')
+')
+
 ifndef(`distro_redhat',`
 	optional_policy(`
 		auth_role(sysadm_r, sysadm_t)
@@ -448,5 +468,34 @@
 	optional_policy(`
 		java_role(sysadm_r, sysadm_t)
 	')
+
+	optional_policy(`
+		gorg_role(sysadm_r, sysadm_t)
+	')
+
+	optional_policy(`
+		skype_role(sysadm_r, sysadm_t)
+	')
+
+	optional_policy(`
+		links_role(sysadm_r, sysadm_t)
+	')
+
+	optional_policy(`
+		mutt_role(sysadm_r, sysadm_t)
+	')
+
+	optional_policy(`
+		qemu_domtrans(sysadm_t)
+		qemu_read_state(sysadm_t)
+		qemu_signal(sysadm_t)
+		qemu_kill(sysadm_t)
+		qemu_setsched(sysadm_t)
+		qemu_run(sysadm_t, sysadm_r)
+	')
+
+	optional_policy(`
+		vde_admin(sysadm_r, sysadm_t)
+	')
 ')
 
