--- refpolicy-20110726/policy/modules/system/selinuxutil.te	2011-07-26 14:10:40.000000000 +0200
+++ refpolicy/policy/modules/system/selinuxutil.te	2011-07-27 08:26:40.201004817 +0200
@@ -199,6 +199,10 @@
 	')
 ')
 
+tunable_policy(`gentoo_try_dontaudit',`
+	gentoo_portage_dontaudit_use_fds(load_policy_t)
+')
+
 ########################################
 #
 # Newrole local policy
@@ -402,6 +406,7 @@
 	ifdef(`distro_gentoo',`
 		# Gentoo integrated run_init:
 		init_script_file_entry_type(run_init_t)
+		gentoo_init_rc_exec(run_init_t)
 	')
 ')
 
@@ -493,6 +498,10 @@
 	')
 ')
 
+optional_policy(`
+	gentoo_portage_eselect_module(semanage_t)
+')
+
 ########################################
 #
 # Setfiles local policy
