--- refpolicy.orig/policy/modules/apps/mozilla.te	2010-12-13 15:11:01.000000000 +0100
+++ refpolicy/policy/modules/apps/mozilla.te	2011-01-02 22:01:27.535999986 +0100
@@ -33,6 +33,10 @@
 files_tmpfs_file(mozilla_tmpfs_t)
 ubac_constrained(mozilla_tmpfs_t)
 
+type mozilla_tmp_t;
+files_tmp_file(mozilla_tmp_t)
+ubac_constrained(mozilla_tmp_t)
+
 ########################################
 #
 # Local policy
@@ -68,6 +72,12 @@
 manage_sock_files_pattern(mozilla_t, mozilla_tmpfs_t, mozilla_tmpfs_t)
 fs_tmpfs_filetrans(mozilla_t, mozilla_tmpfs_t, { file lnk_file sock_file fifo_file })
 
+manage_files_pattern(mozilla_t, mozilla_tmp_t, mozilla_tmp_t)
+manage_dirs_pattern(mozilla_t, mozilla_tmp_t, mozilla_tmp_t)
+files_tmp_filetrans(mozilla_t, mozilla_tmp_t, { file dir } )
+userdom_manage_user_tmp_files(mozilla_t)
+userdom_manage_user_tmp_sockets(mozilla_t)
+
 kernel_read_kernel_sysctls(mozilla_t)
 kernel_read_network_state(mozilla_t)
 # Access /proc, sysctl
--- refpolicy.orig/policy/modules/apps/mozilla.fc	2010-08-03 15:11:03.000000000 +0200
+++ refpolicy/policy/modules/apps/mozilla.fc	2011-01-02 22:23:34.407000019 +0100
@@ -27,3 +27,12 @@
 /usr/lib(64)?/[^/]*firefox[^/]*/firefox-bin -- gen_context(system_u:object_r:mozilla_exec_t,s0)
 /usr/lib/[^/]*firefox[^/]*/firefox --	gen_context(system_u:object_r:mozilla_exec_t,s0)
 /usr/lib64/[^/]*firefox[^/]*/firefox -- gen_context(system_u:object_r:mozilla_exec_t,s0)
+
+ifdef(`distro_gentoo',`
+/usr/bin/firefox-bin	--	gen_context(system_u:object_r:mozilla_exec_t,s0)
+/opt/firefox/libxul\.so	--	gen_context(system_u:object_r:textrel_shlib_t,s0)
+/opt/firefox/firefox	--	gen_context(system_u:object_r:mozilla_exec_t,s0)
+/opt/firefox/run-mozilla.sh	--	gen_context(system_u:object_r:mozilla_exec_t,s0)
+/opt/firefox/firefox-bin	--	gen_context(system_u:object_r:mozilla_exec_t,s0)
+/opt/firefox/plugin-container	--	gen_context(system_u:object_r:mozilla_exec_t,s0)
+')
