--- refpolicy-20110726/policy/modules/kernel/corecommands.fc	2011-06-08 14:38:22.000000000 +0200
+++ refpolicy/policy/modules/kernel/corecommands.fc	2011-08-14 12:13:16.354000324 +0200
@@ -226,6 +241,15 @@
 /usr/lib(64)?/rpm/rpmv		-- 	gen_context(system_u:object_r:bin_t,s0)
 /usr/lib(64)?/sftp-server	--	gen_context(system_u:object_r:bin_t,s0)
 /usr/lib(64)?/vte/gnome-pty-helper --	gen_context(system_u:object_r:bin_t,s0)
+/usr/lib(64)?/xfce4/session/xfsm-shutdown-helper	--	gen_context(system_u:object_r:bin_t,s0)
+/usr/lib(64)?/xfce4/session/balou-export-theme	--	gen_context(system_u:object_r:bin_t,s0)
+/usr/lib(64)?/xfce4/session/balou-install-theme	--	gen_context(system_u:object_r:bin_t,s0)
+/usr/lib(64)?/xfce4/xfwm4/helper-dialog	--	gen_context(system_u:object_r:bin_t,s0)
+/usr/lib(64)?/xfce4/xfconf/xfconfd	--	gen_context(system_u:object_r:bin_t,s0)
+/usr/lib(64)?/xfce4/panel/wrapper	--	gen_context(system_u:object_r:bin_t,s0)
+/usr/lib(64)?/xfce4/panel/migrate	--	gen_context(system_u:object_r:bin_t,s0)
+/usr/lib(64)?/xfce4/exo-1/exo-helper-1	--	gen_context(system_u:object_r:bin_t,s0)
+/usr/lib(64)?/xfce4/exo-1/exo-compose-mail-1	--	gen_context(system_u:object_r:bin_t,s0)
 
 /usr/lib(64)?/debug/bin(/.*)?	--	gen_context(system_u:object_r:bin_t,s0)
 /usr/lib(64)?/debug/sbin(/.*)? --	gen_context(system_u:object_r:bin_t,s0)
