--- refpolicy/policy/modules/apps/gpg.if	2011-03-28 17:05:13.000000000 +0200
+++ refpolicy/policy/modules/apps/gpg.if	2011-09-03 14:13:32.851002302 +0200
@@ -87,6 +87,24 @@
 
 ########################################
 ## <summary>
+##	Execute the gpg application(s) without transitioning
+## </summary>
+## <param name="domain">
+##	<summary>
+##	Domain allowed to execute gpg
+##	</summary>
+## </param>
+#
+interface(`gpg_exec',`
+	gen_require(`
+		type gpg_exec_t;
+	')
+
+	can_exec($1, gpg_exec_t)
+')
+
+########################################
+## <summary>
 ##	Send generic signals to user gpg processes.
 ## </summary>
 ## <param name="domain">
