--- refpolicy-20110726/policy/modules/apps/mozilla.te	2011-07-26 14:10:39.000000000 +0200
+++ refpolicy/policy/modules/apps/mozilla.te	2011-08-09 23:05:18.552001714 +0200
@@ -65,6 +65,10 @@
 # Browse the web, connect to printer
 allow mozilla_t self:tcp_socket create_socket_perms;
 allow mozilla_t self:netlink_route_socket r_netlink_socket_perms;
+# Make sure plugin works
+allow mozilla_t mozilla_plugin_t:process { rlimitinh siginh noatsecure };
+allow mozilla_t mozilla_plugin_t:fd { use };
+allow mozilla_t mozilla_plugin_t:unix_stream_socket { read write };
 
 # for bash - old mozilla binary
 can_exec(mozilla_t, mozilla_exec_t)
@@ -113,6 +117,7 @@
 corenet_tcp_sendrecv_squid_port(mozilla_t)
 corenet_tcp_sendrecv_ftp_port(mozilla_t)
 corenet_tcp_sendrecv_ipp_port(mozilla_t)
+corenet_tcp_sendrecv_tor_port(mozilla_t)
 corenet_tcp_connect_http_port(mozilla_t)
 corenet_tcp_connect_http_cache_port(mozilla_t)
 corenet_tcp_connect_squid_port(mozilla_t)
@@ -120,12 +125,14 @@
 corenet_tcp_connect_ipp_port(mozilla_t)
 corenet_tcp_connect_generic_port(mozilla_t)
 corenet_tcp_connect_soundd_port(mozilla_t)
+corenet_tcp_connect_tor_port(mozilla_t)
 corenet_sendrecv_http_client_packets(mozilla_t)
 corenet_sendrecv_http_cache_client_packets(mozilla_t)
 corenet_sendrecv_squid_client_packets(mozilla_t)
 corenet_sendrecv_ftp_client_packets(mozilla_t)
 corenet_sendrecv_ipp_client_packets(mozilla_t)
 corenet_sendrecv_generic_client_packets(mozilla_t)
+corenet_sendrecv_tor_client_packets(mozilla_t)
 # Should not need other ports
 corenet_dontaudit_tcp_sendrecv_generic_port(mozilla_t)
 corenet_dontaudit_tcp_bind_generic_port(mozilla_t)
@@ -297,7 +305,7 @@
 #
 
 dontaudit mozilla_plugin_t self:capability { sys_ptrace };
-allow mozilla_plugin_t self:process { setsched signal_perms execmem };
+allow mozilla_plugin_t self:process { getsched setsched signal_perms execmem };
 allow mozilla_plugin_t self:fifo_file manage_fifo_file_perms;
 allow mozilla_plugin_t self:unix_stream_socket { connectto create_stream_socket_perms };
 allow mozilla_plugin_t self:tcp_socket create_stream_socket_perms;
@@ -307,8 +315,11 @@
 allow mozilla_plugin_t self:sem create_sem_perms;
 allow mozilla_plugin_t self:shm create_shm_perms;
 
+allow mozilla_plugin_t mozilla_t:unix_stream_socket { read write };
+
 can_exec(mozilla_plugin_t, mozilla_home_t)
-read_files_pattern(mozilla_plugin_t, mozilla_home_t, mozilla_home_t)
+manage_dirs_pattern(mozilla_plugin_t, mozilla_home_t, mozilla_home_t)
+manage_files_pattern(mozilla_plugin_t, mozilla_home_t, mozilla_home_t)
 
 manage_dirs_pattern(mozilla_plugin_t, mozilla_plugin_tmp_t, mozilla_plugin_tmp_t)
 manage_files_pattern(mozilla_plugin_t, mozilla_plugin_tmp_t, mozilla_plugin_tmp_t)
@@ -322,6 +333,7 @@
 manage_sock_files_pattern(mozilla_plugin_t, mozilla_plugin_tmpfs_t, mozilla_plugin_tmpfs_t)
 fs_tmpfs_filetrans(mozilla_plugin_t, mozilla_plugin_tmpfs_t, { file lnk_file sock_file fifo_file })
 
+
 can_exec(mozilla_plugin_t, mozilla_exec_t)
 
 kernel_read_kernel_sysctls(mozilla_plugin_t)
@@ -393,6 +405,9 @@
 userdom_read_user_home_content_files(mozilla_plugin_t)
 userdom_read_user_home_content_symlinks(mozilla_plugin_t)
 
+
+xserver_user_x_domain_template(mozilla_plugin_t, mozilla_plugin_t, mozilla_plugin_tmpfs_t)
+
 tunable_policy(`allow_execmem',`
 	allow mozilla_plugin_t self:process { execmem execstack };
 ')
--- refpolicy-20110726/policy/modules/apps/mozilla.fc	2011-03-28 17:05:13.000000000 +0200
+++ refpolicy/policy/modules/apps/mozilla.fc	2011-08-07 21:59:41.922001218 +0200
@@ -12,6 +12,7 @@
 /usr/bin/mozilla-snapshot	--	gen_context(system_u:object_r:mozilla_exec_t,s0)
 /usr/bin/epiphany-bin		--	gen_context(system_u:object_r:mozilla_exec_t,s0)
 /usr/bin/epiphany		--	gen_context(system_u:object_r:mozilla_exec_t,s0)
+/usr/bin/firefox-bin		--	gen_context(system_u:object_r:mozilla_exec_t,s0)
 /usr/bin/mozilla-[0-9].*	--	gen_context(system_u:object_r:mozilla_exec_t,s0)
 /usr/bin/mozilla-bin-[0-9].*	--	gen_context(system_u:object_r:mozilla_exec_t,s0)
 
@@ -27,3 +28,15 @@
 /usr/lib(64)?/[^/]*firefox[^/]*/firefox-bin -- gen_context(system_u:object_r:mozilla_exec_t,s0)
 /usr/lib/[^/]*firefox[^/]*/firefox --	gen_context(system_u:object_r:mozilla_exec_t,s0)
 /usr/lib64/[^/]*firefox[^/]*/firefox -- gen_context(system_u:object_r:mozilla_exec_t,s0)
+/usr/lib/[^/]*firefox[^/]*/plugin-container	--	gen_context(system_u:object_r:mozilla_plugin_exec_t,s0)
+/usr/lib64/[^/]*firefox[^/]*/plugin-container	--	gen_context(system_u:object_r:mozilla_plugin_exec_t,s0)
+
+
+#
+# /opt
+#
+/opt/firefox/libxul\.so		--	gen_context(system_u:object_r:textrel_shlib_t,s0)
+/opt/firefox/firefox		--	gen_context(system_u:object_r:mozilla_exec_t,s0)
+/opt/firefox/run-mozilla\.sh	--	gen_context(system_u:object_r:mozilla_exec_t,s0)
+/opt/firefox/firefox-bin	--	gen_context(system_u:object_r:mozilla_exec_t,s0)
+/opt/firefox/plugin-container	--	gen_context(system_u:object_r:mozilla_plugin_exec_t,s0)
