--- refpolicy/policy/modules/services/courier.fc	2011-08-19 18:31:17.370999981 +0200
+++ refpolicy/policy/modules/services/courier.fc	2011-08-19 18:29:13.699999981 +0200
@@ -1,4 +1,5 @@
 /etc/courier(/.*)?				gen_context(system_u:object_r:courier_etc_t,s0)
+/etc/courier-imap(/.*)?				gen_context(system_u:object_r:courier_etc_t,s0)
 
 /usr/bin/imapd				--	gen_context(system_u:object_r:courier_pop_exec_t,s0)
 
@@ -23,6 +24,7 @@
 ')
 
 /var/lib/courier(/.*)?				gen_context(system_u:object_r:courier_var_lib_t,s0)
+/var/lib/courier-imap(/.*)?			gen_context(system_u:object_r:courier_var_lib_t,s0)
 
 /var/run/courier(/.*)?				gen_context(system_u:object_r:courier_var_run_t,s0)
 
