--- refpolicy/policy/modules/admin/portage.te	2011-10-17 21:20:50.220000106 +0200
+++ refpolicy/policy/modules/admin/portage.te	2011-10-17 21:19:58.670000105 +0200
@@ -115,7 +115,7 @@
 domain_use_interactive_fds(gcc_config_t)
 
 files_manage_etc_files(gcc_config_t)
-files_rw_etc_runtime_files(gcc_config_t)
+files_manage_etc_runtime_files(gcc_config_t)
 files_read_usr_files(gcc_config_t)
 files_search_var_lib(gcc_config_t)
 files_search_pids(gcc_config_t)
