--- refpolicy/policy/modules/system/init.te	2011-11-03 16:32:08.456865732 +0100
+++ refpolicy/policy/modules/system/init.te	2011-11-03 16:31:14.598865738 +0100
@@ -74,6 +74,9 @@
 type initrc_tmp_t;
 files_tmp_file(initrc_tmp_t)
 
+type initrc_var_log_t;
+logging_log_file(initrc_var_log_t)
+
 type initrc_var_run_t;
 files_pid_file(initrc_var_run_t)
 
@@ -253,6 +256,9 @@
 manage_lnk_files_pattern(initrc_t, initrc_tmp_t, initrc_tmp_t)
 files_tmp_filetrans(initrc_t, initrc_tmp_t, { file dir })
 
+manage_dirs_pattern(initrc_t, initrc_var_log_t, initrc_var_log_t)
+logging_log_filetrans(initrc_t, initrc_var_log_t, dir);
+
 init_write_initctl(initrc_t)
 
 kernel_read_system_state(initrc_t)
