--- refpolicy/policy/modules/services/nagios.te	2011-11-07 20:36:52.627999997 +0100
+++ refpolicy/policy/modules/services/nagios.te	2011-11-07 20:35:50.187999995 +0100
@@ -193,8 +193,9 @@
 manage_files_pattern(nrpe_t, nrpe_var_run_t, nrpe_var_run_t)
 files_pid_filetrans(nrpe_t, nrpe_var_run_t, file)
 
-kernel_read_system_state(nrpe_t)
 kernel_read_kernel_sysctls(nrpe_t)
+kernel_read_software_raid_state(nrpe_t)
+kernel_read_system_state(nrpe_t)
 
 corecmd_exec_bin(nrpe_t)
 corecmd_exec_shell(nrpe_t)
