--- refpolicy/policy/modules/apps/mozilla.te	2012-02-07 20:10:40.240929067 +0100
+++ refpolicy/policy/modules/apps/mozilla.te	2012-02-07 20:10:11.086929005 +0100
@@ -176,10 +176,6 @@
 userdom_use_user_ptys(mozilla_t)
 
 
-xserver_user_x_domain_template(mozilla, mozilla_t, mozilla_tmpfs_t)
-xserver_dontaudit_read_xdm_tmp_files(mozilla_t)
-xserver_dontaudit_getattr_xdm_tmp_sockets(mozilla_t)
-
 tunable_policy(`allow_execmem',`
 	allow mozilla_t self:process { execmem execstack };
 ')
@@ -305,6 +301,12 @@
 	xdg_read_generic_data_home_files(mozilla_t)
 ')
 
+optional_policy(`
+	xserver_user_x_domain_template(mozilla, mozilla_t, mozilla_tmpfs_t)
+	xserver_dontaudit_read_xdm_tmp_files(mozilla_t)
+	xserver_dontaudit_getattr_xdm_tmp_sockets(mozilla_t)
+')
+
 ########################################
 #
 # mozilla_plugin local policy
@@ -412,8 +414,6 @@
 userdom_read_user_home_content_symlinks(mozilla_plugin_t)
 
 
-xserver_user_x_domain_template(mozilla_plugin_t, mozilla_plugin_t, mozilla_plugin_tmpfs_t)
-
 tunable_policy(`allow_execmem',`
 	allow mozilla_plugin_t self:process { execmem execstack };
 ')
@@ -477,4 +477,5 @@
 	xserver_read_xdm_pid(mozilla_plugin_t)
 	xserver_stream_connect(mozilla_plugin_t)
 	xserver_use_user_fonts(mozilla_plugin_t)
+	xserver_user_x_domain_template(mozilla_plugin_t, mozilla_plugin_t, mozilla_plugin_tmpfs_t)
 ')
