diff -uNr refpolicy/policy/modules/contrib/mozilla.te refpolicy/policy/modules/contrib/mozilla.te
--- refpolicy/policy/modules/contrib/mozilla.te	2012-07-26 21:09:59.539963767 +0200
+++ refpolicy/policy/modules/contrib/mozilla.te	2012-07-26 21:09:59.732963767 +0200
@@ -168,6 +168,7 @@
 
 mozilla_run_plugin(mozilla_t, mozilla_roles)
 
+
 xserver_user_x_domain_template(mozilla, mozilla_t, mozilla_tmpfs_t)
 xserver_dontaudit_read_xdm_tmp_files(mozilla_t)
 xserver_dontaudit_getattr_xdm_tmp_sockets(mozilla_t)
@@ -292,6 +293,10 @@
 	thunderbird_domtrans(mozilla_t)
 ')
 
+optional_policy(`
+	xdg_read_generic_data_home_files(mozilla_t)
+')
+
 ########################################
 #
 # mozilla_plugin local policy
