diff -uNr refpolicy/policy/modules/contrib/gpg.te refpolicy/policy/modules/contrib/gpg.te
--- refpolicy/policy/modules/contrib/gpg.te	2012-07-26 21:09:59.538963767 +0200
+++ refpolicy/policy/modules/contrib/gpg.te	2012-07-26 21:09:59.934963767 +0200
@@ -340,6 +340,12 @@
 ')
 
 optional_policy(`
+	mutt_read_home_files(gpg_t)
+	mutt_read_tmp_files(gpg_t)
+	mutt_rw_tmp_files(gpg_t)
+')
+
+optional_policy(`
 	pulseaudio_exec(gpg_pinentry_t)
 	pulseaudio_rw_home_files(gpg_pinentry_t)
 	pulseaudio_setattr_home_dir(gpg_pinentry_t)
