diff -uNr refpolicy/policy/modules/contrib/pan.fc refpolicy/policy/modules/contrib/pan.fc
--- refpolicy/policy/modules/contrib/pan.fc	1970-01-01 01:00:00.000000000 +0100
+++ refpolicy/policy/modules/contrib/pan.fc	2012-07-26 21:10:00.824963769 +0200
@@ -0,0 +1,6 @@
+HOME_DIR/\.pan2(/.*)?		gen_context(system_u:object_r:pan_home_t,s0)
+
+#
+# /usr
+#
+/usr/bin/pan		--	gen_context(system_u:object_r:pan_exec_t,s0)
diff -uNr refpolicy/policy/modules/contrib/pan.if refpolicy/policy/modules/contrib/pan.if
--- refpolicy/policy/modules/contrib/pan.if	1970-01-01 01:00:00.000000000 +0100
+++ refpolicy/policy/modules/contrib/pan.if	2012-07-26 21:10:00.824963769 +0200
@@ -0,0 +1,38 @@
+## <summary>Pan news reader client</summary>
+
+########################################
+## <summary>
+##	Role access for pan
+## </summary>
+## <param name="role">
+##	<summary>
+##	Role allowed access
+##	</summary>
+## </param>
+## <param name="domain">
+##	<summary>
+##	User domain for the role
+##	</summary>
+## </param>
+#
+interface(`pan_role',`
+	gen_require(`
+		type pan_t, pan_exec_t, pan_home_t;
+	')
+	role $1 types pan_t;
+
+	allow $2 pan_t:process signal_perms;
+
+	domtrans_pattern($2, pan_exec_t, pan_t)
+
+	ps_process_pattern($2, pan_t)
+
+        manage_dirs_pattern($2, pan_home_t, pan_home_t)
+        manage_files_pattern($2, pan_home_t, pan_home_t)
+        manage_lnk_files_pattern($2, pan_home_t, pan_home_t)
+
+        relabel_dirs_pattern($2, pan_home_t, pan_home_t)
+        relabel_files_pattern($2, pan_home_t, pan_home_t)
+        relabel_lnk_files_pattern($2, pan_home_t, pan_home_t)
+')
+
diff -uNr refpolicy/policy/modules/contrib/pan.te refpolicy/policy/modules/contrib/pan.te
--- refpolicy/policy/modules/contrib/pan.te	1970-01-01 01:00:00.000000000 +0100
+++ refpolicy/policy/modules/contrib/pan.te	2012-07-26 21:10:00.824963769 +0200
@@ -0,0 +1,116 @@
+policy_module(pan, 1.0)
+
+########################################
+#
+# Declarations
+#
+
+## <desc>
+## <p>
+##   Be able to manage user files (needed to support sending and downloading
+##   attachments). Without this boolean set, only files marked as pan_home_t
+##   can be used for sending and receiving.
+## </p>
+## </desc>
+gen_tunable(pan_manage_user_content, false)
+
+
+type pan_t;
+type pan_exec_t;
+application_domain(pan_t, pan_exec_t)
+ubac_constrained(pan_t)
+
+type pan_home_t;
+userdom_user_home_content(pan_home_t)
+
+type pan_tmpfs_t;
+files_tmpfs_file(pan_tmpfs_t)
+ubac_constrained(pan_tmpfs_t)
+
+########################################
+#
+# Pan local policy
+#
+allow pan_t self:process { getsched signal };
+allow pan_t self:fifo_file rw_fifo_file_perms;
+allow pan_t pan_tmpfs_t:file { read write };
+
+# Allow pan to work with its ~/.pan2 location
+manage_dirs_pattern(pan_t, pan_home_t, pan_home_t)
+manage_files_pattern(pan_t, pan_home_t, pan_home_t)
+manage_lnk_files_pattern(pan_t, pan_home_t, pan_home_t)
+
+# Support for shared memory
+fs_tmpfs_filetrans(pan_t, pan_tmpfs_t, file)
+
+kernel_dontaudit_read_system_state(pan_t)
+
+corenet_all_recvfrom_netlabel(pan_t)
+corenet_all_recvfrom_unlabeled(pan_t)
+corenet_sendrecv_innd_client_packets(pan_t)
+corenet_tcp_connect_innd_port(pan_t)
+corenet_tcp_sendrecv_generic_if(pan_t)
+corenet_tcp_sendrecv_generic_node(pan_t)
+corenet_tcp_sendrecv_innd_port(pan_t)
+
+domain_dontaudit_use_interactive_fds(pan_t)
+
+files_read_etc_files(pan_t)
+files_read_usr_files(pan_t)
+
+miscfiles_read_localization(pan_t)
+
+sysnet_dns_name_resolve(pan_t)
+
+userdom_dontaudit_use_user_ttys(pan_t)
+userdom_use_user_ptys(pan_t)
+
+xserver_user_x_domain_template(pan, pan_t, pan_tmpfs_t)
+
+tunable_policy(`pan_manage_user_content',`
+	userdom_manage_user_home_content_dirs(pan_t)
+	userdom_manage_user_home_content_files(pan_t)
+')
+
+tunable_policy(`use_nfs_home_dirs',`
+	fs_manage_nfs_dirs(pan_t)
+	fs_manage_nfs_files(pan_t)
+	fs_manage_nfs_symlinks(pan_t)
+')
+
+tunable_policy(`use_samba_home_dirs',`
+	fs_manage_cifs_dirs(pan_t)
+	fs_manage_cifs_files(pan_t)
+	fs_manage_cifs_symlinks(pan_t)
+')
+
+optional_policy(`
+	cups_read_rw_config(pan_t)
+')
+
+optional_policy(`
+	dbus_system_bus_client(pan_t)
+	dbus_session_bus_client(pan_t)
+')
+
+optional_policy(`
+	gnome_stream_connect_gconf(pan_t)
+')
+
+optional_policy(`
+	gpg_domtrans(pan_t)
+	gpg_signal(pan_t)
+')
+
+optional_policy(`
+	lpd_domtrans_lpr(pan_t)
+')
+
+optional_policy(`
+	mozilla_read_user_home_files(pan_t)
+	mozilla_domtrans(pan_t)
+')
+
+optional_policy(`
+	xdg_read_generic_data_home_files(pan_t)
+')
