diff -uNr refpolicy/policy/modules/roles/unprivuser.te refpolicy/policy/modules/roles/unprivuser.te
--- refpolicy/policy/modules/roles/unprivuser.te	2012-07-26 21:10:01.019963769 +0200
+++ refpolicy/policy/modules/roles/unprivuser.te	2012-07-26 21:10:01.154963771 +0200
@@ -86,6 +86,10 @@
 	')
 
 	optional_policy(`
+		links_role(user_r, user_t)
+	')
+
+	optional_policy(`
 		lockdev_role(user_r, user_t)
 	')
 
