diff -uNr refpolicy/policy/modules/roles/staff.te refpolicy/policy/modules/roles/staff.te
--- refpolicy/policy/modules/roles/staff.te	2012-07-26 21:10:01.152963771 +0200
+++ refpolicy/policy/modules/roles/staff.te	2012-07-26 21:10:01.288963771 +0200
@@ -153,6 +153,10 @@
 	')
 
 	optional_policy(`
+		skype_role(staff_r, staff_t)
+	')
+
+	optional_policy(`
 		screen_role_template(staff, staff_r, staff_t)
 	')
 
