diff -uNr refpolicy/policy/modules/roles/sysadm.te refpolicy/policy/modules/roles/sysadm.te
--- refpolicy/policy/modules/roles/sysadm.te	2012-07-26 21:09:59.525963767 +0200
+++ refpolicy/policy/modules/roles/sysadm.te	2012-07-26 21:10:01.553963771 +0200
@@ -463,6 +463,10 @@
 	')
 
 	optional_policy(`
+		gorg_role(sysadm_r, sysadm_t)
+	')
+
+	optional_policy(`
 		gpg_role(sysadm_r, sysadm_t)
 	')
 
