diff -uNr refpolicy/policy/modules/system/logging.fc refpolicy/policy/modules/system/logging.fc
--- refpolicy/policy/modules/system/logging.fc	2012-07-26 21:10:02.087963773 +0200
+++ refpolicy/policy/modules/system/logging.fc	2012-07-26 21:10:03.027963775 +0200
@@ -24,6 +24,7 @@
 /usr/sbin/syslog-ng	--	gen_context(system_u:object_r:syslogd_exec_t,s0)
 /usr/sbin/syslogd	--	gen_context(system_u:object_r:syslogd_exec_t,s0)
 
+/var/lib/misc/syslog-ng\.persist-	--	gen_context(system_u:object_r:syslogd_var_lib_t,s0)
 /var/lib/syslog-ng(/.*)? 	gen_context(system_u:object_r:syslogd_var_lib_t,s0)
 /var/lib/r?syslog(/.*)?		gen_context(system_u:object_r:syslogd_var_lib_t,s0)
 /var/lib/syslog-ng.persist --	gen_context(system_u:object_r:syslogd_var_lib_t,s0)
