diff -uNr refpolicy/policy/modules/kernel/files.if refpolicy/policy/modules/kernel/files.if
--- refpolicy/policy/modules/kernel/files.if	2012-07-26 21:10:02.157963773 +0200
+++ refpolicy/policy/modules/kernel/files.if	2012-07-26 21:10:04.846963777 +0200
@@ -6218,6 +6218,26 @@
 
 ########################################
 ## <summary>
+##	Create PID directories.
+## </summary>
+## <param name="domain">
+##	<summary>
+##	Domain allowed access.
+##	</summary>
+## </param>
+#
+interface(`files_create_pid_dirs',`
+	gen_require(`
+		type var_t, var_run_t;
+	')
+
+	allow $1 var_t:dir search_dir_perms;
+	allow $1 var_run_t:lnk_file read_lnk_file_perms;
+	create_dirs_pattern($1, var_run_t, var_run_t)
+')
+
+########################################
+## <summary>
 ##	Delete all process IDs.
 ## </summary>
 ## <param name="domain">
