diff -uNr refpolicy/policy/modules/system/init.te refpolicy/policy/modules/system/init.te
--- refpolicy/policy/modules/system/init.te	2012-07-26 21:10:05.043963779 +0200
+++ refpolicy/policy/modules/system/init.te	2012-07-26 21:10:05.115963779 +0200
@@ -361,6 +361,8 @@
 files_mounton_isid_type_dirs(initrc_t)
 files_list_default(initrc_t)
 files_mounton_default(initrc_t)
+files_manage_generic_tmp_files(initrc_t)
+files_manage_generic_tmp_dirs(initrc_t)
 
 fs_write_cgroup_files(initrc_t)
 fs_list_inotifyfs(initrc_t)
