diff -uNr refpolicy/policy/modules/system/udev.fc refpolicy/policy/modules/system/udev.fc
--- refpolicy/policy/modules/system/udev.fc	2012-07-26 21:09:59.527963767 +0200
+++ refpolicy/policy/modules/system/udev.fc	2012-07-26 21:10:06.899963782 +0200
@@ -9,6 +9,7 @@
 /etc/udev/rules.d(/.*)? gen_context(system_u:object_r:udev_rules_t,s0)
 /etc/udev/scripts/.+ --	gen_context(system_u:object_r:udev_helper_exec_t,s0)
 
+/lib/udev/udevd	--	gen_context(system_u:object_r:udev_exec_t,s0)
 /lib/udev/udev-acl --	gen_context(system_u:object_r:udev_exec_t,s0)
 
 ifdef(`distro_debian',`
@@ -26,6 +27,7 @@
 /sbin/start_udev --	gen_context(system_u:object_r:udev_exec_t,s0)
 ')
 
+/usr/bin/udevadm	--	gen_context(system_u:object_r:udev_exec_t,s0)
 /usr/bin/udevinfo --	gen_context(system_u:object_r:udev_exec_t,s0)
 
 /var/run/PackageKit/udev(/.*)? gen_context(system_u:object_r:udev_var_run_t,s0)
