diff -uNr refpolicy/policy/modules/kernel/files.if refpolicy/policy/modules/kernel/files.if
--- refpolicy/policy/modules/kernel/files.if	2012-07-26 21:10:04.910963779 +0200
+++ refpolicy/policy/modules/kernel/files.if	2012-07-26 21:10:08.782963786 +0200
@@ -2594,6 +2594,25 @@
 
 ########################################
 ## <summary>
+##	Do not audit attempts to read files
+##	in /etc 
+## </summary>
+## <param name="domain">
+##	<summary>
+##	Domain to not audit.
+##	</summary>
+## </param>
+#
+interface(`files_dontaudit_read_etc_files',`
+	gen_require(`
+		type etc_runtime_t;
+	')
+
+	dontaudit $1 etc_t:file { getattr read };
+')
+
+########################################
+## <summary>
 ##	Read generic files in /etc.
 ## </summary>
 ## <desc>
@@ -2908,6 +2927,24 @@
 ')
 
 ########################################
+## <summary>
+##	Do not audit attempts to read etc_runtime resources
+## </summary>
+## <param name="domain">
+##	<summary>
+##	Domain allowed access.
+##	</summary>
+## </param>
+#
+interface(`files_dontaudit_read_etc_runtime',`
+	gen_require(`
+		type etc_runtime_t;
+	')
+
+	dontaudit $1 etc_runtime_t:file read_file_perms;
+')
+
+########################################
 ## <summary>
 ##	Read files in /etc that are dynamically
 ##	created on boot, such as mtab.
