--- refpolicy/policy/modules/admin/portage.te	2011-09-03 14:58:48.981002380 +0200
+++ refpolicy/policy/modules/admin/portage.te	2011-09-03 14:57:08.803002377 +0200
@@ -73,6 +73,9 @@
 type portage_cache_t;
 files_type(portage_cache_t)
 
+type portage_gpg_t;
+files_type(portage_gpg_t)
+
 type portage_log_t;
 logging_log_file(portage_log_t)
 
@@ -329,6 +332,10 @@
 	fs_manage_nfs_symlinks(portage_fetch_t)
 ')
 
+optional_policy(`
+	gpg_exec(portage_fetch_t)
+')
+
 ##########################################
 #
 # Portage sandbox domain
--- refpolicy/policy/modules/admin/portage.fc	2011-09-03 14:58:48.962002380 +0200
+++ refpolicy/policy/modules/admin/portage.fc	2011-09-03 14:47:42.660002361 +0200
@@ -1,6 +1,7 @@
 /etc/make\.conf			--	gen_context(system_u:object_r:portage_conf_t,s0)
 /etc/make\.globals		--	gen_context(system_u:object_r:portage_conf_t,s0)
 /etc/portage(/.*)?			gen_context(system_u:object_r:portage_conf_t,s0)
+/etc/portage/gpg(/.*)?			gen_context(system_u:object_r:portage_gpg_t,s0)
 
 /usr/bin/gcc-config		--	gen_context(system_u:object_r:gcc_config_exec_t,s0)
 /usr/bin/layman			--	gen_context(system_u:object_r:portage_fetch_exec_t,s0)
