--- refpolicy/policy/modules/admin/portage.if	2011-09-03 13:30:09.067002228 +0200
+++ refpolicy/policy/modules/admin/portage.if	2011-09-03 13:29:44.786002227 +0200
@@ -16,6 +16,7 @@
 interface(`portage_domtrans',`
 	gen_require(`
 		type portage_t, portage_exec_t;
+		type portage_fetch_t, portage_fetch_exec_t;
 	')
 
 	files_search_usr($1)
@@ -23,6 +24,7 @@
 
 	# transition to portage
 	domtrans_pattern($1, portage_exec_t, portage_t)
+	domtrans_pattern($1, portage_fetch_exec_t, portage_fetch_t)
 ')
 
 ########################################
