--- refpolicy-20110726/policy/modules/system/mount.te	2011-07-26 14:10:41.000000000 +0200
+++ refpolicy/policy/modules/system/mount.te	2011-07-27 08:19:43.312004818 +0200
@@ -141,6 +142,12 @@
 	')
 ')
 
+ifdef(`distro_gentoo',`
+	tunable_policy(`gentoo_try_dontaudit',`
+		term_dontaudit_manage_pty_dirs(mount_t)
+	')
+')
+
 tunable_policy(`allow_mount_anyfile',`
 	auth_read_all_dirs_except_auth_files(mount_t)
 	auth_read_all_files_except_auth_files(mount_t)
