--- refpolicy/policy/modules/contrib/fail2ban.te	2012-02-25 11:58:56.447321105 +0100
+++ refpolicy/policy/modules/contrib/fail2ban.te	2012-02-25 11:58:43.789321079 +0100
@@ -50,6 +50,9 @@
 manage_files_pattern(fail2ban_t, fail2ban_var_run_t, fail2ban_var_run_t)
 files_pid_filetrans(fail2ban_t, fail2ban_var_run_t, { dir file sock_file })
 
+# FAM support needs this (/proc/self and parent stuff)
+read_files_pattern(fail2ban_t, fail2ban_t, fail2ban_t)
+
 kernel_read_system_state(fail2ban_t)
 
 corecmd_exec_bin(fail2ban_t)
