--- refpolicy/policy/modules/services/postfix.te	2011-12-28 17:56:03.513213162 +0100
+++ refpolicy/policy/modules/services/postfix.te	2011-12-28 17:55:12.472213053 +0100
@@ -93,7 +93,7 @@
 #
 
 # chown is to set the correct ownership of queue dirs
-allow postfix_master_t self:capability { chown dac_override kill setgid setuid net_bind_service sys_tty_config dac_read_search };
+allow postfix_master_t self:capability { chown dac_override kill setgid setuid net_bind_service sys_tty_config dac_read_search fowner fsetid };
 allow postfix_master_t self:fifo_file rw_fifo_file_perms;
 allow postfix_master_t self:tcp_socket create_stream_socket_perms;
 allow postfix_master_t self:udp_socket create_socket_perms;
