--- refpolicy/policy/modules/contrib/mozilla.te	2012-02-25 12:54:46.856328256 +0100
+++ refpolicy/policy/modules/contrib/mozilla.te	2012-02-25 12:54:32.329328226 +0100
@@ -301,6 +301,7 @@
 ')
 
 optional_policy(`
+	xdg_read_generic_config_home_files(mozilla_t)
 	xdg_read_generic_data_home_files(mozilla_t)
 ')
 
@@ -469,6 +470,10 @@
 ')
 
 optional_policy(`
+	xdg_read_generic_config_home_files(mozilla_plugin_t)
+')
+
+optional_policy(`
 	xserver_read_xdm_pid(mozilla_plugin_t)
 	xserver_stream_connect(mozilla_plugin_t)
 	xserver_use_user_fonts(mozilla_plugin_t)
