--- refpolicy/policy/modules/system/unconfined.te	2011-09-09 21:11:47.938007866 +0200
+++ refpolicy/policy/modules/system/unconfined.te	2011-09-09 21:09:44.981007854 +0200
@@ -53,6 +53,7 @@
 	seutil_run_runinit(unconfined_t, unconfined_r)
 	seutil_init_script_run_runinit(unconfined_t, unconfined_r)
 	portage_run(unconfined_t, unconfined_r)
+	portage_run_fetch(unconfined_t, unconfined_r)
 	portage_run_gcc_config(unconfined_t, unconfined_r)
 ')
 
