--- refpolicy/policy/modules/admin/portage.fc	2011-09-21 21:00:11.847001630 +0200
+++ refpolicy/policy/modules/admin/portage.fc	2011-09-21 20:58:57.378001626 +0200
@@ -4,6 +4,7 @@
 /etc/portage/gpg(/.*)?			gen_context(system_u:object_r:portage_gpg_t,s0)
 
 /usr/bin/gcc-config		--	gen_context(system_u:object_r:gcc_config_exec_t,s0)
+/usr/bin/glsa-check		--	gen_context(system_u:object_r:portage_exec_t,s0)
 /usr/bin/layman			--	gen_context(system_u:object_r:portage_fetch_exec_t,s0)
 /usr/bin/sandbox		--	gen_context(system_u:object_r:portage_exec_t,s0)
 
