--- refpolicy/policy/modules/services/amavis.fc	2011-03-21 15:13:58.000000000 +0100
+++ refpolicy/policy/modules/services/amavis.fc	2011-11-05 17:28:09.835849120 +0100
@@ -1,5 +1,6 @@
 
 /etc/amavis\.conf		--	gen_context(system_u:object_r:amavis_etc_t,s0)
+/etc/amavisd\.conf		--	gen_context(system_u:object_r:amavis_etc_t,s0)
 /etc/amavisd(/.*)?			gen_context(system_u:object_r:amavis_etc_t,s0)
 /etc/rc\.d/init\.d/amavis	--	gen_context(system_u:object_r:amavis_initrc_exec_t,s0)
 
