--- refpolicy/policy/modules/system/init.fc	2012-03-03 20:19:29.187745366 +0100
+++ refpolicy/policy/modules/system/init.fc	2012-03-03 20:18:56.117745296 +0100
@@ -26,8 +26,7 @@
 #
 ifdef(`distro_gentoo', `
 /lib/rc/init\.d(/.*)?		gen_context(system_u:object_r:initrc_state_t,s0)
-/lib32/rc/init\.d(/.*)?		gen_context(system_u:object_r:initrc_state_t,s0)
-/lib64/rc/init\.d(/.*)?		gen_context(system_u:object_r:initrc_state_t,s0)
+/lib/rc/console(/.*)?		gen_context(system_u:object_r:initrc_state_t,s0)
 ')
 
 #
