--- refpolicy/policy/modules/kernel/selinux.if	2011-09-26 15:58:28.000000000 +0200
+++ refpolicy/policy/modules/kernel/selinux.if	2012-03-05 20:30:46.976003009 +0100
@@ -63,6 +63,10 @@
 	# (/selinux) is already a selinuxfs
 	allow $1 security_t:filesystem getattr;
 
+	# Same for /sys/fs/selinux
+	dev_getattr_sysfs_fs($1)
+	dev_search_sysfs($1)
+
 	# read /proc/filesystems to see if selinuxfs is supported
 	# then read /proc/self/mount to see where selinuxfs is mounted
 	kernel_read_system_state($1)
@@ -89,6 +93,10 @@
 	# (/selinux) is already a selinuxfs
 	dontaudit $1 security_t:filesystem getattr;
 
+	# Same for /sys/fs/selinux
+	dev_dontaudit_getattr_sysfs_fs($1)
+	dev_dontaudit_search_sysfs($1)
+
 	# read /proc/filesystems to see if selinuxfs is supported
 	# then read /proc/self/mount to see where selinuxfs is mounted
 	kernel_dontaudit_read_system_state($1)
@@ -165,6 +173,8 @@
 	')
 
 	allow $1 security_t:filesystem getattr;
+	dev_getattr_sysfs_fs($1)
+	dev_search_sysfs($1)
 ')
 
 ########################################
@@ -184,6 +194,9 @@
 	')
 
 	dontaudit $1 security_t:filesystem getattr;
+
+	dev_dontaudit_getattr_sysfs_fs($1)
+	dev_dontaudit_search_sysfs($1)
 ')
 
 ########################################
