--- refpolicy/policy/modules/services/inetd.te	2010-08-03 15:11:06.000000000 +0200
+++ refpolicy/policy/modules/services/inetd.te	2011-11-11 15:57:50.209547053 +0100
@@ -38,9 +38,9 @@
 # Local policy
 #
 
-allow inetd_t self:capability { setuid setgid };
+allow inetd_t self:capability { setuid setgid sys_resource };
 dontaudit inetd_t self:capability sys_tty_config;
-allow inetd_t self:process { setsched setexec };
+allow inetd_t self:process { setsched setexec setrlimit };
 allow inetd_t self:fifo_file rw_fifo_file_perms;
 allow inetd_t self:tcp_socket create_stream_socket_perms;
 allow inetd_t self:udp_socket create_socket_perms;
