--- refpolicy/policy/modules/apps/links.te	2011-11-13 10:43:11.479875607 +0100
+++ refpolicy/policy/modules/apps/links.te	2011-11-13 10:42:13.216875482 +0100
@@ -5,18 +5,22 @@
 # Declarations
 #
 
+## <desc>
+## <p>
+## Allow links to manage files in users home directories (download files)
+## </p>
+## </desc>
+gen_tunable(links_manage_user_files, false)
+
 type links_t;
 type links_exec_t;
-application_domain(links_t, links_exec_t)
+userdom_user_application_domain(links_t, links_exec_t)
 
 type links_home_t;
-typealias links_home_t alias { user_links_home_t staff_links_home_t sysadm_links_home_t };
 userdom_user_home_content(links_home_t)
 
 type links_tmpfs_t;
-typealias links_tmpfs_t alias { user_links_tmpfs_t staff_links_tmpfs_t sysadm_links_tmpfs_t };
-files_tmpfs_file(links_tmpfs_t)
-ubac_constrained(links_tmpfs_t)
+userdom_user_tmpfs_file(links_tmpfs_t)
 
 ############################
 #
@@ -44,9 +48,6 @@
 
 domain_use_interactive_fds(links_t)
 
-userdom_manage_user_home_content_dirs(links_t)
-userdom_manage_user_home_content_files(links_t)
-
 auth_use_nsswitch(links_t)
 
 userdom_use_user_terminals(links_t)
@@ -55,6 +56,12 @@
 
 miscfiles_read_localization(links_t)
 
+tunable_policy(`links_manage_user_files',`
+	userdom_manage_user_home_content_dirs(links_t)
+	userdom_manage_user_home_content_files(links_t)
+')
+
+
 optional_policy(`
 	xserver_user_x_domain_template(links, links_t, links_tmpfs_t)
 ')
