--- refpolicy/policy/modules/kernel/storage.fc	2011-03-31 15:13:41.000000000 +0200
+++ refpolicy/policy/modules/kernel/storage.fc	2011-11-15 10:30:56.792001319 +0100
@@ -1,4 +1,4 @@
-
+/dev/\.tmp-block.*	-c	gen_context(system_u:object_r:fixed_disk_device_t,mls_systemhigh)
 /dev/n?(raw)?[qr]ft[0-3] -c	gen_context(system_u:object_r:tape_device_t,s0)
 /dev/n?[hs]t[0-9].*	-c	gen_context(system_u:object_r:tape_device_t,s0)
 /dev/n?z?qft[0-3]	-c	gen_context(system_u:object_r:tape_device_t,s0)
