--- refpolicy/policy/modules/system/logging.if	2011-11-15 15:48:28.000000000 +0100
+++ refpolicy/policy/modules/system/logging.if	2012-03-13 20:52:20.931460371 +0100
@@ -604,6 +604,25 @@
 
 ########################################
 ## <summary>
+##	Delete the syslog socket files
+## </summary>
+## <param name="domain">
+##	<summary>
+##	Domain allowed access
+##	</summary>
+## </param>
+## <rolecap/>
+#
+interface(`logging_delete_devlog_socket',`
+	gen_require(`
+		type devlog_t;
+	')
+
+	allow $1 devlog_t:sock_file unlink;
+')
+
+########################################
+## <summary>
 ##	Allows the domain to open a file in the
 ##	log directory, but does not allow the listing
 ##	of the contents of the log directory.
