--- refpolicy/policy/modules/admin/sudo.if	2011-03-28 17:05:13.000000000 +0200
+++ refpolicy/policy/modules/admin/sudo.if	2011-12-06 10:18:31.056019952 +0100
@@ -137,6 +137,7 @@
 	userdom_use_user_terminals($1_sudo_t)
 	# for some PAM modules and for cwd
 	userdom_dontaudit_search_user_home_content($1_sudo_t)
+	userdom_dontaudit_search_user_home_dirs($1_sudo_t)
 
 	ifdef(`hide_broken_symptoms', `
 		dontaudit $1_sudo_t $3:socket_class_set { read write };
