--- refpolicy/policy/modules/system/init.te	2012-03-25 16:24:05.813350979 +0200
+++ refpolicy/policy/modules/system/init.te	2012-03-25 16:23:29.319350901 +0200
@@ -223,7 +223,7 @@
 #
 
 allow initrc_t self:process { getpgid setsched setpgid setrlimit getsched };
-allow initrc_t self:capability ~{ sys_admin sys_module };
+allow initrc_t self:capability ~{ sys_module };
 dontaudit initrc_t self:capability sys_module; # sysctl is triggering this
 allow initrc_t self:passwd rootok;
 allow initrc_t self:key manage_key_perms;
