--- refpolicy/policy/modules/apps/mozilla.te	2011-12-31 14:08:25.598737257 +0100
+++ refpolicy/policy/modules/apps/mozilla.te	2011-12-31 14:07:54.133737191 +0100
@@ -157,6 +157,7 @@
 files_read_var_symlinks(mozilla_t)
 files_dontaudit_getattr_boot_dirs(mozilla_t)
 
+fs_dontaudit_getattr_all_fs(mozilla_t)
 fs_search_auto_mountpoints(mozilla_t)
 fs_list_inotifyfs(mozilla_t)
 fs_rw_tmpfs_files(mozilla_t)
