--- refpolicy/policy/modules/kernel/devices.if	2012-05-16 22:25:58.403360926 +0200
+++ refpolicypolicy/modules/kernel/devices.if	2012-05-16 22:20:09.722360182 +0200
@@ -588,6 +588,25 @@
 
 ########################################
 ## <summary>
+##	Setattr for generic character device files.
+## </summary>
+## <param name="domain">
+##	<summary>
+##	Domain allowed access
+##	</summary>
+## </param>
+#
+interface(`dev_setattr_generic_chr_files',`
+	gen_require(`
+		type device_t;
+	')
+
+	allow $1 device_t:chr_file setattr;
+')
+
+
+########################################
+## <summary>
 ##	Read generic character device files.
 ## </summary>
 ## <param name="domain">
