--- refpolicy/policy/modules/contrib/awstats.te	2011-10-25 19:19:57.000000000 +0200
+++ refpolicypolicy/modules/contrib/awstats.te	2012-06-08 21:07:06.050005943 +0200
@@ -17,7 +17,6 @@
 type awstats_var_lib_t;
 files_type(awstats_var_lib_t)
 
-apache_content_template(awstats)
 
 ########################################
 #
@@ -59,6 +58,7 @@
 
 sysnet_dns_name_resolve(awstats_t)
 
+apache_content_template(awstats)
 apache_read_log(awstats_t)
 
 optional_policy(`
