--- refpolicy/policy/modules/services/rpc.te	2011-07-26 14:10:40.000000000 +0200
+++ refpolicy/policy/modules/services/rpc.te	2011-08-18 23:17:32.453000050 +0200
@@ -112,7 +112,7 @@
 #
 
 allow nfsd_t self:capability { dac_override dac_read_search sys_admin sys_resource };
-
+allow nfsd_t self:udp_socket listen;
 allow nfsd_t exports_t:file read_file_perms;
 allow nfsd_t { nfsd_rw_t nfsd_ro_t }:dir list_dir_perms;
 
--- refpolicy/policy/modules/services/rpc.if	2011-07-22 13:03:41.000000000 +0200
+++ refpolicy/policy/modules/services/rpc.if	2012-01-29 14:33:02.840215098 +0100
@@ -431,5 +431,6 @@
 	')
 
 	files_search_var_lib($1)
+	rw_dirs_pattern($1, var_lib_nfs_t, var_lib_nfs_t)
 	manage_files_pattern($1, var_lib_nfs_t, var_lib_nfs_t)
 ')
