--- refpolicy/policy/modules/roles/sysadm.te	2012-06-26 21:26:52.577499718 +0200
+++ refpolicypolicy/modules/roles/sysadm.te	2012-04-21 20:07:09.819210979 +0200
@@ -249,6 +249,10 @@
 ')
 
 optional_policy(`
+	networkmanager_run_wpa_cli(sysadm_t, sysadm_r)
+')
+
+optional_policy(`
 	netutils_run(sysadm_t, sysadm_r)
 	netutils_run_ping(sysadm_t, sysadm_r)
 	netutils_run_traceroute(sysadm_t, sysadm_r)
