--- refpolicy/policy/modules/system/miscfiles.te	2011-10-25 19:19:35.000000000 +0200
+++ refpolicypolicy/modules/system/miscfiles.te	2012-07-26 15:40:31.254921571 +0200
@@ -14,6 +14,13 @@
 miscfiles_cert_type(cert_t)
 
 #
+# cert_home_t is the type of files in the users' home directories.
+#
+type cert_home_t;
+miscfiles_cert_type(cert_home_t)
+userdom_user_home_content(cert_home_t)
+
+#
 # fonts_t is the type of various font
 # files in /usr
 #
--- refpolicy/policy/modules/system/miscfiles.if	2011-10-25 19:19:35.000000000 +0200
+++ refpolicypolicy/modules/system/miscfiles.if	2012-07-26 16:08:26.122925145 +0200
@@ -171,6 +171,35 @@
 
 ########################################
 ## <summary>
+##	Automatically use the cert_home_t label for selected resources created
+##	in a users home directory
+## </summary>
+## <param name="domain">
+##	<summary>
+##	Domain allowed access
+##	</summary>
+## </param>
+## <param name="class">
+##	<summary>
+##	Resource type(s) for which the label should be used
+##	</summary>
+## </param>
+## <param name="filename" optional="true">
+##	<summary>
+##	Name of the resource that is being created
+##	</summary>
+## </param>
+#
+interface(`miscfiles_user_home_dir_filetrans_cert_home',`
+	gen_require(`
+		type cert_home_t;
+	')
+
+	userdom_user_home_dir_filetrans($1, cert_home_t, $2, $3)
+')
+
+########################################
+## <summary>
 ##	Read fonts.
 ## </summary>
 ## <param name="domain">
--- refpolicy/policy/modules/system/miscfiles.fc	2012-07-26 16:09:00.519925221 +0200
+++ refpolicypolicy/modules/system/miscfiles.fc	2012-07-26 16:00:06.758924078 +0200
@@ -91,3 +91,5 @@
 /var/empty/sshd/etc/localtime -- gen_context(system_u:object_r:locale_t,s0)
 /var/spool/postfix/etc/localtime -- gen_context(system_u:object_r:locale_t,s0)
 ')
+
+HOME_DIR/.nss(/.*)?		gen_context(system_u:object_r:cert_home_t)
