--- refpolicy.orig/policy/modules/services/ldap.te	2010-09-10 17:05:45.000000000 +0200
+++ refpolicy/policy/modules/services/ldap.te	2011-03-12 22:10:48.814999997 +0100
@@ -42,11 +42,12 @@
 # cjp: why net_raw?
 allow slapd_t self:capability { kill setgid setuid net_raw dac_override dac_read_search };
 dontaudit slapd_t self:capability sys_tty_config;
-allow slapd_t self:process setsched;
+allow slapd_t self:process { setsched signal };
 allow slapd_t self:fifo_file rw_fifo_file_perms;
 allow slapd_t self:udp_socket create_socket_perms;
 #slapd needs to listen and accept needed by ldapsearch (slapd needs to accept from ldapseach)
 allow slapd_t self:tcp_socket create_stream_socket_perms;
+allow slapd_t self:unix_stream_socket listen;
 
 allow slapd_t slapd_cert_t:dir list_dir_perms;
 read_files_pattern(slapd_t, slapd_cert_t, slapd_cert_t)
@@ -114,6 +115,7 @@
 
 userdom_dontaudit_use_unpriv_user_fds(slapd_t)
 userdom_dontaudit_search_user_home_dirs(slapd_t)
+userdom_use_user_terminals(slapd_t)
 
 optional_policy(`
 	kerberos_keytab_template(slapd, slapd_t)
--- refpolicy.orig/policy/modules/services/ldap.if	2010-08-03 15:11:06.000000000 +0200
+++ refpolicy/policy/modules/services/ldap.if	2011-03-12 22:10:01.379999997 +0100
@@ -94,7 +94,7 @@
 	gen_require(`
 		type slapd_t, slapd_tmp_t, slapd_replog_t;
 		type slapd_lock_t, slapd_etc_t, slapd_var_run_t;
-		type slapd_initrc_exec_t;
+		type slapd_initrc_exec_t, slapd_exec_t;
 	')
 
 	allow $1 slapd_t:process { ptrace signal_perms };
@@ -105,6 +105,9 @@
 	role_transition $2 slapd_initrc_exec_t system_r;
 	allow $2 system_r;
 
+	role $2 types { slapd_exec_t slapd_t };
+	domtrans_pattern($1, slapd_exec_t, slapd_t)
+
 	files_list_etc($1)
 	admin_pattern($1, slapd_etc_t)
 
--- refpolicy.orig/policy/modules/services/ldap.fc	2010-08-03 15:11:06.000000000 +0200
+++ refpolicy/policy/modules/services/ldap.fc	2011-03-12 18:57:10.880999997 +0100
@@ -8,7 +8,16 @@
 /usr/lib/slapd		--	gen_context(system_u:object_r:slapd_exec_t,s0)
 ')
 
+ifdef(`distro_gentoo',`
+/usr/lib(64)?/openldap/slapd	--	gen_context(system_u:object_r:slapd_exec_t,s0)
+')
+
 /var/lib/ldap(/.*)?		gen_context(system_u:object_r:slapd_db_t,s0)
+ifdef(`distro_gentoo',`
+/var/lib/openldap-data(/.*)?	gen_context(system_u:object_r:slapd_db_t,s0)
+/var/lib/openldap-ldbm(/.*)?	gen_context(system_u:object_r:slapd_db_t,s0)
+/var/lib/openldap-slurpd(/.*)?	gen_context(system_u:object_r:slapd_db_t,s0)
+')
 /var/lib/ldap/replog(/.*)?	gen_context(system_u:object_r:slapd_replog_t,s0)
 
 /var/run/ldapi		-s	gen_context(system_u:object_r:slapd_var_run_t,s0)
