--- refpolicy.orig/policy/modules/roles/sysadm.te	2010-12-13 15:11:01.000000000 +0100
+++ refpolicy/policy/modules/roles/sysadm.te	2011-07-24 17:58:46.413000793 +0200
@@ -34,6 +34,7 @@
 ubac_fd_exempt(sysadm_t)
 
 init_exec(sysadm_t)
+gentoo_init_rc_exec(sysadm_t)
 
 # Add/remove user home directories
 userdom_manage_user_home_dirs(sysadm_t)
@@ -48,6 +49,10 @@
 		optional_policy(`
 			seutil_init_script_run_runinit(sysadm_t, sysadm_r)
 		')
+
+		#optional_policy(`
+		#	gentoo_init_initrc_exec_notrans(sysadm, sysadm_r, sysadm_t)
+		#')
 	')
 ')
 
@@ -174,6 +179,10 @@
 ')
 
 optional_policy(`
+	ldap_admin(sysadm_t, sysadm_r)
+')
+
+optional_policy(`
 	libs_run_ldconfig(sysadm_t, sysadm_r)
 ')
 
@@ -261,7 +270,7 @@
 ')
 
 optional_policy(`
-	raid_domtrans_mdadm(sysadm_t)
+	gentoo_raid_mdadm_role(sysadm_r, sysadm_t)
 ')
 
 optional_policy(`
@@ -400,6 +409,26 @@
 	yam_run(sysadm_t, sysadm_r)
 ')
 
+optional_policy(`
+	networkmanager_run_wpa_cli(sysadm_t, sysadm_r)
+')
+
+optional_policy(`
+	postfix_run(master, sysadm_t, sysadm_r)
+	postfix_run(postqueue, sysadm_t, sysadm_r)
+')
+
+ifdef(`distro_gentoo', `
+	optional_policy(`
+		postgresql_admin(sysadm_t, sysadm_r)
+		postgresql_domtrans(sysadm_t)
+	')
+
+	optional_policy(`
+		nginx_admin(sysadm_t, sysadm_r)
+	')
+')
+
 ifndef(`distro_redhat',`
 	optional_policy(`
 		auth_role(sysadm_r, sysadm_t)
@@ -448,5 +477,38 @@
 	optional_policy(`
 		java_role(sysadm_r, sysadm_t)
 	')
+
+	optional_policy(`
+		gorg_role(sysadm_r, sysadm_t)
+	')
+
+	optional_policy(`
+		skype_role(sysadm_r, sysadm_t)
+	')
+
+	optional_policy(`
+		links_role(sysadm_r, sysadm_t)
+	')
+
+	optional_policy(`
+		mutt_role(sysadm_r, sysadm_t)
+	')
+
+	optional_policy(`
+		pan_role(sysadm_r, sysadm_t)
+	')
+
+	optional_policy(`
+		qemu_domtrans(sysadm_t)
+		qemu_read_state(sysadm_t)
+		qemu_signal(sysadm_t)
+		qemu_kill(sysadm_t)
+		qemu_setsched(sysadm_t)
+		qemu_run(sysadm_t, sysadm_r)
+	')
+
+	optional_policy(`
+		vde_admin(sysadm_r, sysadm_t)
+	')
 ')
 
