--- refpolicy.orig/policy/modules/services/postgresql.if	2010-08-05 15:25:29.000000000 +0200
+++ refpolicy/policy/modules/services/postgresql.if	2011-02-13 14:23:07.247905045 +0100
@@ -431,6 +431,8 @@
 
 	typeattribute $1 sepgsql_admin_type;
 
+	role $2 types postgresql_t;
+
 	allow $1 postgresql_t:process { ptrace signal_perms };
 	ps_process_pattern($1, postgresql_t)
 
--- refpolicy.orig/policy/modules/services/postgresql.fc	2010-08-03 15:11:07.000000000 +0200
+++ refpolicy/policy/modules/services/postgresql.fc	2011-02-13 13:40:48.798905046 +0100
@@ -5,6 +5,10 @@
 /etc/rc\.d/init\.d/(se)?postgresql --	gen_context(system_u:object_r:postgresql_initrc_exec_t,s0)
 /etc/sysconfig/pgsql(/.*)? 		gen_context(system_u:object_r:postgresql_etc_t,s0)
 
+ifdef(`distro_gentoo', `
+/etc/postgresql-.*(/.*)?		gen_context(system_u:object_r:postgresql_etc_t,s0)
+')
+
 #
 # /usr
 #
@@ -23,6 +27,10 @@
 /usr/share/jonas/pgsql(/.*)?		gen_context(system_u:object_r:postgresql_db_t,s0)
 ')
 
+ifdef(`distro_gentoo', `
+/usr/lib(64)?/postgresql-.*/bin/.*	--	gen_context(system_u:object_r:postgresql_exec_t,s0)
+')
+
 #
 # /var
 #
